<!DOCTYPE html>
<html lang="en">
<head>
<!-- 2020-05-08 Fri 11:22 -->
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<title>Snakemake Pipeline for LD score Regression</title>
<meta name="generator" content="Org mode">
<meta name="author" content="Nicholas Knoblauch">
<style type="text/css">
 <!--/*--><![CDATA[/*><!--*/
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #ccc;
    box-shadow: 3px 3px 3px #eee;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: visible;
    padding-top: 1.2em;
  }
  pre.src:before {
    display: none;
    position: absolute;
    background-color: white;
    top: -10px;
    right: 10px;
    padding: 3px;
    border: 1px solid black;
  }
  pre.src:hover:before { display: inline;}
  /* Languages per Org manual */
  pre.src-asymptote:before { content: 'Asymptote'; }
  pre.src-awk:before { content: 'Awk'; }
  pre.src-C:before { content: 'C'; }
  /* pre.src-C++ doesn't work in CSS */
  pre.src-clojure:before { content: 'Clojure'; }
  pre.src-css:before { content: 'CSS'; }
  pre.src-D:before { content: 'D'; }
  pre.src-ditaa:before { content: 'ditaa'; }
  pre.src-dot:before { content: 'Graphviz'; }
  pre.src-calc:before { content: 'Emacs Calc'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-fortran:before { content: 'Fortran'; }
  pre.src-gnuplot:before { content: 'gnuplot'; }
  pre.src-haskell:before { content: 'Haskell'; }
  pre.src-hledger:before { content: 'hledger'; }
  pre.src-java:before { content: 'Java'; }
  pre.src-js:before { content: 'Javascript'; }
  pre.src-latex:before { content: 'LaTeX'; }
  pre.src-ledger:before { content: 'Ledger'; }
  pre.src-lisp:before { content: 'Lisp'; }
  pre.src-lilypond:before { content: 'Lilypond'; }
  pre.src-lua:before { content: 'Lua'; }
  pre.src-matlab:before { content: 'MATLAB'; }
  pre.src-mscgen:before { content: 'Mscgen'; }
  pre.src-ocaml:before { content: 'Objective Caml'; }
  pre.src-octave:before { content: 'Octave'; }
  pre.src-org:before { content: 'Org mode'; }
  pre.src-oz:before { content: 'OZ'; }
  pre.src-plantuml:before { content: 'Plantuml'; }
  pre.src-processing:before { content: 'Processing.js'; }
  pre.src-python:before { content: 'Python'; }
  pre.src-R:before { content: 'R'; }
  pre.src-ruby:before { content: 'Ruby'; }
  pre.src-sass:before { content: 'Sass'; }
  pre.src-scheme:before { content: 'Scheme'; }
  pre.src-screen:before { content: 'Gnu Screen'; }
  pre.src-sed:before { content: 'Sed'; }
  pre.src-sh:before { content: 'shell'; }
  pre.src-sql:before { content: 'SQL'; }
  pre.src-sqlite:before { content: 'SQLite'; }
  /* additional languages in org.el's org-babel-load-languages alist */
  pre.src-forth:before { content: 'Forth'; }
  pre.src-io:before { content: 'IO'; }
  pre.src-J:before { content: 'J'; }
  pre.src-makefile:before { content: 'Makefile'; }
  pre.src-maxima:before { content: 'Maxima'; }
  pre.src-perl:before { content: 'Perl'; }
  pre.src-picolisp:before { content: 'Pico Lisp'; }
  pre.src-scala:before { content: 'Scala'; }
  pre.src-shell:before { content: 'Shell Script'; }
  pre.src-ebnf2ps:before { content: 'ebfn2ps'; }
  /* additional language identifiers per "defun org-babel-execute"
       in ob-*.el */
  pre.src-cpp:before  { content: 'C++'; }
  pre.src-abc:before  { content: 'ABC'; }
  pre.src-coq:before  { content: 'Coq'; }
  pre.src-groovy:before  { content: 'Groovy'; }
  /* additional language identifiers from org-babel-shell-names in
     ob-shell.el: ob-shell is the only babel language using a lambda to put
     the execution function name together. */
  pre.src-bash:before  { content: 'bash'; }
  pre.src-csh:before  { content: 'csh'; }
  pre.src-ash:before  { content: 'ash'; }
  pre.src-dash:before  { content: 'dash'; }
  pre.src-ksh:before  { content: 'ksh'; }
  pre.src-mksh:before  { content: 'mksh'; }
  pre.src-posh:before  { content: 'posh'; }
  /* Additional Emacs modes also supported by the LaTeX listings package */
  pre.src-ada:before { content: 'Ada'; }
  pre.src-asm:before { content: 'Assembler'; }
  pre.src-caml:before { content: 'Caml'; }
  pre.src-delphi:before { content: 'Delphi'; }
  pre.src-html:before { content: 'HTML'; }
  pre.src-idl:before { content: 'IDL'; }
  pre.src-mercury:before { content: 'Mercury'; }
  pre.src-metapost:before { content: 'MetaPost'; }
  pre.src-modula-2:before { content: 'Modula-2'; }
  pre.src-pascal:before { content: 'Pascal'; }
  pre.src-ps:before { content: 'PostScript'; }
  pre.src-prolog:before { content: 'Prolog'; }
  pre.src-simula:before { content: 'Simula'; }
  pre.src-tcl:before { content: 'tcl'; }
  pre.src-tex:before { content: 'TeX'; }
  pre.src-plain-tex:before { content: 'Plain TeX'; }
  pre.src-verilog:before { content: 'Verilog'; }
  pre.src-vhdl:before { content: 'VHDL'; }
  pre.src-xml:before { content: 'XML'; }
  pre.src-nxml:before { content: 'XML'; }
  /* add a generic configuration mode; LaTeX export needs an additional
     (add-to-list 'org-latex-listings-langs '(conf " ")) in .emacs */
  pre.src-conf:before { content: 'Configuration File'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  .org-svg { width: 90%; }
  /*]]>*/-->
</style>
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css" integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm" crossorigin="anonymous">
<link rel="stylesheet" type="text/css" href="https://gongzhitaao.org/orgcss/org.css"/>
<style>.src-python {background: #3f3f3f; color: #dcdccc;}</style>
<style>.src-snakemake {background: #3f3f3f; color: #dcdccc;}</style>
<style>.src-R {background: #3f3f3f; color: #dcdccc;}</style>
<script type="text/javascript">
/*
@licstart  The following is the entire license notice for the
JavaScript code in this tag.

Copyright (C) 2012-2019 Free Software Foundation, Inc.

The JavaScript code in this tag is free software: you can
redistribute it and/or modify it under the terms of the GNU
General Public License (GNU GPL) as published by the Free Software
Foundation, either version 3 of the License, or (at your option)
any later version.  The code is distributed WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS
FOR A PARTICULAR PURPOSE.  See the GNU GPL for more details.

As additional permission under GNU GPL version 3 section 7, you
may distribute non-source (e.g., minimized or compacted) forms of
that code without the copy of the GNU GPL normally required by
section 4, provided you include this license notice and a URL
through which recipients can access the Corresponding Source.


@licend  The above is the entire license notice
for the JavaScript code in this tag.
*/
<!--/*--><![CDATA[/*><!--*/
 function CodeHighlightOn(elem, id)
 {
   var target = document.getElementById(id);
   if(null != target) {
     elem.cacheClassElem = elem.className;
     elem.cacheClassTarget = target.className;
     target.className = "code-highlighted";
     elem.className   = "code-highlighted";
   }
 }
 function CodeHighlightOff(elem, id)
 {
   var target = document.getElementById(id);
   if(elem.cacheClassElem)
     elem.className = elem.cacheClassElem;
   if(elem.cacheClassTarget)
     target.className = elem.cacheClassTarget;
 }
/*]]>*///-->
</script>
</head>
<body>
<div id="content">
<h1 class="title">Snakemake Pipeline for LD score Regression</h1>
<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#org3332ea4">Running LDSC</a>
<ul>
<li><a href="#org65baa36">Downloading data</a></li>
<li><a href="#org9e505f2">Preprocessing</a></li>
<li><a href="#org47333f8">Computing LD scores and estimating heritability</a></li>
</ul>
</li>
</ul>
</div>
</div>

<div id="outline-container-org3332ea4" class="outline-2">
<h2 id="org3332ea4">Running LDSC</h2>
<div class="outline-text-2" id="text-org3332ea4">
<p>
I have my own fork of ldsc that formats output so that it's easier to read.   <a href="https://github.com/xinhe-lab/ldsc/">It can be found here</a>
</p>
</div>

<div id="outline-container-org65baa36" class="outline-3">
<h3 id="org65baa36">Downloading data</h3>
<div class="outline-text-3" id="text-org65baa36">
<p>
We'll need the baseline LD scores as well as the reference panel so we can compute our 
own LD scores on the same set of SNPs and samples with different annotations.  For working with other annotations,
(including PTB related annotations) look <a href="https://crerecombinase.github.io/ptb_workflowr/annotation.html">here</a>.
</p>


<p>
The tangled file: <code>../workflow/dl_ldsc_snakefile</code> can be found here: <a href="https://github.com/ptb_workflowr/tree/master/workflow/dl_ldsc_snakefile">https://github.com/ptb_workflowr/tree/master/workflow/dl_ldsc_snakefile</a>
</p>
<div class="org-src-container">
<pre class="src src-snakemake">
<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">get_baseline_model</span>:
<span style="color: #FFABAB;">"""Download 'baseline' LD model v 2.2"""</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #B46DCC;">temp</span>(config_d[<span style="color: #FFABAB;">'DL'</span>]+<span style="color: #FFABAB;">"1000G_Phase3_baselineLD_v2.2_ldscores.tgz"</span>)
    <span style="color: #99D6FF;">shell</span>:
        <span style="color: #FFABAB;">"wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_baselineLD_v2.2_ldscores.tgz -O {output}"</span>

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">gunzip_baseline</span>:
<span style="color: #FFABAB;">"""unzip baseline v2.2 model ld scores"""</span>    
    <span style="color: #99D6FF;">input</span>:
        config_d[<span style="color: #FFABAB;">'DL'</span>] +<span style="color: #FFABAB;">"1000G_Phase3_baselineLD_v2.2_ldscores.tgz"</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">ldfiles</span> = <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"baseline/baselineLD.{chrom}.l2.ldscore.gz"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23)),
        <span style="color: #FF9326;">annotf</span> = <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"baseline/baselineLD.{chrom}.annot.gz"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23)),
        <span style="color: #FF9326;">m50</span> = <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"baseline/baselineLD.{chrom}.l2.M_5_50"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23))
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">L2</span>=config_d[<span style="color: #FFABAB;">'L2'</span>]
    <span style="color: #99D6FF;">shell</span>:
        <span style="color: #FFABAB;">"tar -xvzf {input} -C {params.L2}/baseline"</span>

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">get_weights</span>:
<span style="color: #FFABAB;">"""Download baseline LDscore weights (needed to run LD score regression)"""</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #B46DCC;">temp</span>(config_d[<span style="color: #FFABAB;">"DL"</span>]+<span style="color: #FFABAB;">"1000G_Phase3_weights_hm3_no_MHC.tgz"</span>)
    <span style="color: #99D6FF;">shell</span>:
        <span style="color: #FFABAB;">"wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_weights_hm3_no_MHC.tgz -O {output}"</span>

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">gunzip_weights</span>:
<span style="color: #FFABAB;">"""unzip baseline weights"""</span>
    <span style="color: #99D6FF;">input</span>:
        config_d[<span style="color: #FFABAB;">"DL"</span>]+<span style="color: #FFABAB;">"1000G_Phase3_weights_hm3_no_MHC.tgz"</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">ldfiles</span> = <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'WEIGHTS'</span>] +<span style="color: #FFABAB;">"weights.hm3_noMHC.{chrom}.l2.ldscore.gz"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23))
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">W</span>=config_d[<span style="color: #FFABAB;">'1KG'</span>]
    <span style="color: #99D6FF;">shell</span>:
        <span style="color: #FFABAB;">"tar -xvzf {input} -C {params.W}"</span>        

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">get_plinkfiles</span>:
<span style="color: #FFABAB;">"""Download tar.gz of (european) reference panel genotypes"""</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #B46DCC;">temp</span>(config_d[<span style="color: #FFABAB;">'DL'</span>] +<span style="color: #FFABAB;">"1000G_Phase3_plinkfiles.tgz"</span>)
    <span style="color: #99D6FF;">shell</span>:
        <span style="color: #FFABAB;">"wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_plinkfiles.tgz -O {output}"</span>

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">gunzip_plinkfiles</span>:
<span style="color: #FFABAB;">"""Unzip downloaded reference panel genotypes"""</span>
    <span style="color: #99D6FF;">input</span>:
        config_d[<span style="color: #FFABAB;">'DL'</span>] +<span style="color: #FFABAB;">"1000G_Phase3_plinkfiles.tgz"</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">fam_files</span> = <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'1KG'</span>] +<span style="color: #FFABAB;">"1000G_EUR_Phase3_plink/1000G.EUR.QC.{chrom}.fam"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23)),
        <span style="color: #FF9326;">bim_files</span> = <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'1KG'</span>] +<span style="color: #FFABAB;">"1000G_EUR_Phase3_plink/1000G.EUR.QC.{chrom}.bim"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23)),
        <span style="color: #FF9326;">bed_files</span> = <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'1KG'</span>] +<span style="color: #FFABAB;">"1000G_EUR_Phase3_plink/1000G.EUR.QC.{chrom}.bed"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23))
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">KG</span>=config_d[<span style="color: #FFABAB;">'1KG'</span>]
    <span style="color: #99D6FF;">shell</span>:
        <span style="color: #FFABAB;">"tar -xvzf {input} -C {params.KG}"</span>

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">get_frq</span>:
<span style="color: #FFABAB;">"""Download allele frequency data for phase 3 genotypes"""</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #B46DCC;">temp</span>(config_d[<span style="color: #FFABAB;">'DL'</span>]+<span style="color: #FFABAB;">"1000G_Phase3_frq.tgz"</span>)
    <span style="color: #99D6FF;">shell</span>:
        <span style="color: #FFABAB;">"wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1000G_Phase3_frq.tgz -O {output}"</span>

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">gunzip_frqf</span>:
<span style="color: #FFABAB;">"""unzip allele frequency data"""</span>
    <span style="color: #99D6FF;">input</span>:
        config_d[<span style="color: #FFABAB;">'DL'</span>] +<span style="color: #FFABAB;">"1000G_Phase3_frq.tgz"</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">fam_files</span> = <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'FRQF'</span>] +<span style="color: #FFABAB;">"1000G.EUR.QC.{chrom}.frq"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23)),
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">KG</span>=config_d[<span style="color: #FFABAB;">'1KG'</span>]
    <span style="color: #99D6FF;">shell</span>:
        <span style="color: #FFABAB;">"tar -xvzf {input} -C {params.KG}"</span>

</pre>
</div>
</div>
</div>

<div id="outline-container-org9e505f2" class="outline-3">
<h3 id="org9e505f2">Preprocessing</h3>
<div class="outline-text-3" id="text-org9e505f2">
<p>
I have added some R scripts to make preparing the data easier. This section of the snakemake pipeline is a combo of my R scripts and ldsc calls for data prep.
</p>


<p>
The tangled file: <code>../workflow/ldsc_snakefile</code> can be found here: <a href="https://github.com/ptb_workflowr/tree/master/workflow/ldsc_snakefile">https://github.com/ptb_workflowr/tree/master/workflow/ldsc_snakefile</a>
</p>
<div class="org-src-container">
<pre class="src src-snakemake"><span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">indexgwas2h5</span>:
    <span style="color: #6E8C02;">"""Starting with a copy of the gwas summary statistics, pull out the subset that match at the SNPs we have baseline LD for"""</span>    
    <span style="color: #99D6FF;">input</span>:
        <span style="color: #FF9326;">inputf</span>=config_d[<span style="color: #FFABAB;">'GWAS'</span>] +<span style="color: #FFABAB;">"{gwas}_gwas.h5"</span>,
        <span style="color: #FF9326;">indexf</span>=config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"baseline/baselineLD.{chrom}.l2.ldscore.gz"</span>
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">chrom</span>=<span style="color: #FFABAB;">"{chrom}"</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">outputf</span>=<span style="color: #B46DCC;">temp</span>(config_d[<span style="color: #FFABAB;">'GWAS'</span>] +<span style="color: #FFABAB;">"hm3_index/{gwas}_gwas_hm_chr{chrom}.tsv"</span>)
    <span style="color: #99D6FF;">conda</span>:
        config_e[<span style="color: #FFABAB;">'r'</span>]
    <span style="color: #99D6FF;">script</span>:
        <span style="color: #FFABAB;">"../scripts/index_gwas.R"</span>

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">make_annot</span>:
    <span style="color: #FFABAB;">"""Create per-annotation annot files for each bed file"""</span>        
    <span style="color: #99D6FF;">input</span>:
        <span style="color: #FF9326;">anno_bed</span>=<span style="color: #B46DCC;">ancient</span>(config_d[<span style="color: #FFABAB;">'BED'</span>] +<span style="color: #FFABAB;">"{annot}.bed"</span>),
        <span style="color: #FF9326;">bim</span>=config_d[<span style="color: #FFABAB;">'1KG'</span>] + <span style="color: #FFABAB;">"1000G_EUR_Phase3_plink/1000G.EUR.QC.{chrom}.bim"</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">annot</span> = config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"{annot}.{chrom}.annot.gz"</span>
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">anno_name</span>=<span style="color: #FFABAB;">'{annot}'</span>
    <span style="color: #99D6FF;">conda</span>:
        config_e[<span style="color: #FFABAB;">'ldsc'</span>]
    <span style="color: #99D6FF;">shell</span>:
        config_d[<span style="color: #FFABAB;">'LDSC'</span>]+<span style="color: #FFABAB;">"make_annot.py --bed-file {input.anno_bed} --bimfile {input.bim} --annot-file {output.annot} --annot-name {params.anno_name}"</span>

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">prep_ldsc_sumstsat</span>:
    <span style="color: #FFABAB;">"""Create ldsc formatted summary statistics (or at least suitable for merge_sumstats.py)"""</span>        
    <span style="color: #99D6FF;">input</span>:
        <span style="color: #FF9326;">inputf</span>=<span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'GWAS'</span>] +<span style="color: #FFABAB;">"hm3_index/{{gwas}}_gwas_hm_chr{chrom}.tsv"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23))
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">gwas_t</span>=<span style="color: #FFABAB;">""</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">outputf</span>=<span style="color: #B46DCC;">temp</span>(config_d[<span style="color: #FFABAB;">'GWAS'</span>] +<span style="color: #FFABAB;">"ldsc_input_pre/{gwas}_gwas.sumstats.gz"</span>)
    <span style="color: #99D6FF;">conda</span>:
        config_e[<span style="color: #FFABAB;">'r'</span>]
    <span style="color: #99D6FF;">script</span>:
        <span style="color: #FFABAB;">"../scripts/gen_ldsc_sumstats.R"</span>

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">check_ldsc_sumstat</span>:
    <span style="color: #FFABAB;">"""Let LDSC do a final pre-processing of the summary statistics"""</span>
    <span style="color: #99D6FF;">input</span>:
        config_d[<span style="color: #FFABAB;">'GWAS'</span>] +<span style="color: #FFABAB;">"ldsc_input_pre/{gwas}_gwas.sumstats.gz"</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">outputf</span>=config_d[<span style="color: #FFABAB;">'GWAS'</span>] +<span style="color: #FFABAB;">"ldsc_input/{gwas}_gwas.sumstats.gz"</span>
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">outputf</span>=config_d[<span style="color: #FFABAB;">'GWAS'</span>] +<span style="color: #FFABAB;">"ldsc_input/{gwas}_gwas"</span>
    <span style="color: #99D6FF;">conda</span>:
        config_e[<span style="color: #FFABAB;">'ldsc'</span>]
    <span style="color: #99D6FF;">log</span>:
        <span style="color: #FF9326;">logf</span>=config_d[<span style="color: #FFABAB;">'GWAS'</span>] +<span style="color: #FFABAB;">"ldsc_input/{gwas}_gwas.log"</span>
    <span style="color: #99D6FF;">shell</span>:
        config_d[<span style="color: #FFABAB;">'LDSC'</span>]+<span style="color: #FFABAB;">"munge_sumstats.py --sumstats {input} --out {params.outputf}"</span>


<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">pull_rsid</span>:
    <span style="color: #FFABAB;">"""Generate a snplist file from an ldscore file"""</span>    
    <span style="color: #99D6FF;">input</span>:        
        config_d[<span style="color: #FFABAB;">"L2"</span>]+<span style="color: #FFABAB;">"baseline/baselineLD.{chrom}.l2.ldscore.gz"</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #B46DCC;">temp</span>(config_d[<span style="color: #FFABAB;">"L2"</span>]+<span style="color: #FFABAB;">"snplist/{chrom}.snplist.txt"</span>)
    <span style="color: #99D6FF;">shell</span>:
        <span style="color: #FFABAB;">"zcat {input} | cut -f 2 | tail -n +2 &gt; {output}"</span>


<span style="color: #5D6A70;"># </span><span style="color: #5D6A70;">TODO remove this</span>
<span style="color: #5D6A70;">#</span><span style="color: #5D6A70;">""" This is an awful hack I came up with so that ld scores don't get recomputed, and I should get rid of it"""</span>
<span style="color: #C8FF03;">def</span> <span style="color: #FFE203;">norr_ldsc</span>(<span style="color: #B46DCC;">wildcards</span>):
    <span style="color: #FF9326;">chrom</span> = <span style="color: #B46DCC;">wildcards</span>.chrom
    <span style="color: #FF9326;">annot</span> = <span style="color: #B46DCC;">wildcards</span>.annot
    <span style="color: #FF9326;">anno_bed</span>=config_d[<span style="color: #FFABAB;">'L2'</span>] +f<span style="color: #FFABAB;">"{annot}.{chrom}.annot.gz"</span>
    <span style="color: #FF9326;">snplistf</span>=config_d[<span style="color: #FFABAB;">"L2"</span>]+f<span style="color: #FFABAB;">"snplist/{chrom}.snplist.txt"</span>
    <span style="color: #FF9326;">bim</span>=config_d[<span style="color: #FFABAB;">'1KG'</span>] + f<span style="color: #FFABAB;">"1000G_EUR_Phase3_plink/1000G.EUR.QC.{chrom}.bim"</span>
    <span style="color: #FF9326;">bed</span>=config_d[<span style="color: #FFABAB;">'1KG'</span>] + f<span style="color: #FFABAB;">"1000G_EUR_Phase3_plink/1000G.EUR.QC.{chrom}.bed"</span>
    <span style="color: #FF9326;">fam</span>=config_d[<span style="color: #FFABAB;">'1KG'</span>] + f<span style="color: #FFABAB;">"1000G_EUR_Phase3_plink/1000G.EUR.QC.{chrom}.fam"</span>
    <span style="color: #FF9326;">l2</span>=(config_d[<span style="color: #FFABAB;">'L2'</span>]+f<span style="color: #FFABAB;">"{annot}.{chrom}.l2.M"</span>)
    <span style="color: #FF9326;">l2M_50</span>=(config_d[<span style="color: #FFABAB;">'L2'</span>]+f<span style="color: #FFABAB;">"{annot}.{chrom}.l2.M_5_50"</span>)
    <span style="color: #FF9326;">l2gz</span>=(config_d[<span style="color: #FFABAB;">'L2'</span>]+f<span style="color: #FFABAB;">"{annot}.{chrom}.l2.ldscore.gz"</span>)
    <span style="color: #C8FF03;">if</span> <span style="color: #B46DCC;">all</span>(os.path.exists(x) <span style="color: #C8FF03;">for</span> x <span style="color: #C8FF03;">in</span> [l2,l2M_50,l2gz]):
        <span style="color: #C8FF03;">return</span> {}
    <span style="color: #C8FF03;">else</span>:
        <span style="color: #C8FF03;">return</span> {<span style="color: #FFABAB;">'anno_bed'</span>: anno_bed,
                <span style="color: #FFABAB;">'snplistf'</span>: snplistf,
                <span style="color: #FFABAB;">'bim'</span>: bim,
                <span style="color: #FFABAB;">'bed'</span>: bed,
                <span style="color: #FFABAB;">'fam'</span>: fam}

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">cmp_ldscores</span>:
    <span style="color: #FFABAB;">""" Compute stratified ld scores for the single annotation at a single chromosome"""</span>
    <span style="color: #99D6FF;">input</span>:
        <span style="color: #B46DCC;">unpack</span>(norr_ldsc)
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">l2</span>=(config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"{annot}.{chrom}.l2.M"</span>),
        <span style="color: #FF9326;">l2M_50</span>=(config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"{annot}.{chrom}.l2.M_5_50"</span>),
        <span style="color: #FF9326;">l2gz</span>=(config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"{annot}.{chrom}.l2.ldscore.gz"</span>)
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">plink</span>=config_d[<span style="color: #FFABAB;">'1KG'</span>] + <span style="color: #FFABAB;">"1000G_EUR_Phase3_plink/1000G.EUR.QC.{chrom}"</span>,
        <span style="color: #FF9326;">odir</span>=config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"{annot}.{chrom}"</span>,
        <span style="color: #FF9326;">anno</span>=<span style="color: #FFABAB;">"{annot}"</span>
    <span style="color: #99D6FF;">wildcard_constraints</span>:
        <span style="color: #FF9326;">annot</span>=<span style="color: #FFABAB;">"[^/]+"</span>
    <span style="color: #99D6FF;">conda</span>:
        config_e[<span style="color: #FFABAB;">'ldsc'</span>]
    <span style="color: #99D6FF;">shell</span>:
        config_d[<span style="color: #FFABAB;">'LDSC'</span>]+<span style="color: #FFABAB;">"ldsc.py --l2 --bfile {params.plink} --print-snps {input.snplistf} --ld-wind-cm 1 --thin-annot --annot {input.anno_bed} --out {params.odir} &amp;&amp; cp {output.l2gz} {output.l2gz}~ &amp;&amp; zcat {output.l2gz}~ | sed '1s/L2/{params.anno}/' | gzip  &gt; {output.l2gz} &amp;&amp; rm {output.l2gz}~"</span>


</pre>
</div>



<p>
The tangled file: <code>../workflow/baseline_ldsc_snakefile</code> can be found here: <a href="https://github.com/ptb_workflowr/tree/master/workflow/baseline_ldsc_snakefile">https://github.com/ptb_workflowr/tree/master/workflow/baseline_ldsc_snakefile</a>
</p>
<div class="org-src-container">
<pre class="src src-snakemake">
<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">cmp_baselineb_ldscores</span>:
    <span style="color: #FFABAB;">""" Compute baseline ld scores for the single annotation at a single chromosome"""</span>
    <span style="color: #99D6FF;">input</span>:
        <span style="color: #FF9326;">bim</span>=config_d[<span style="color: #FFABAB;">'1KG'</span>] + <span style="color: #FFABAB;">"1000G_EUR_Phase3_plink/1000G.EUR.QC.{chrom}.bim"</span>,
        <span style="color: #FF9326;">bed</span>=config_d[<span style="color: #FFABAB;">'1KG'</span>] + <span style="color: #FFABAB;">"1000G_EUR_Phase3_plink/1000G.EUR.QC.{chrom}.bed"</span>,
        <span style="color: #FF9326;">fam</span>=config_d[<span style="color: #FFABAB;">'1KG'</span>] + <span style="color: #FFABAB;">"1000G_EUR_Phase3_plink/1000G.EUR.QC.{chrom}.fam"</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">l2</span>=(config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"true_baseline/ldsc.{chrom}.l2.M"</span>),
        <span style="color: #FF9326;">l2M_50</span>=(config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"true_baseline/ldsc.{chrom}.l2.M_5_50"</span>),
        <span style="color: #FF9326;">l2gz</span>=(config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"true_baseline/ldsc.{chrom}.l2.ldscore.gz"</span>)
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">plink</span>=config_d[<span style="color: #FFABAB;">'1KG'</span>] + <span style="color: #FFABAB;">"1000G_EUR_Phase3_plink/1000G.EUR.QC.{chrom}"</span>,
        <span style="color: #FF9326;">odir</span>=config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"true_baseline/ldsc.{chrom}"</span>
    <span style="color: #99D6FF;">conda</span>:
        config_e[<span style="color: #FFABAB;">'ldsc'</span>]
    <span style="color: #99D6FF;">shell</span>:
        config_d[<span style="color: #FFABAB;">'LDSC'</span>]+<span style="color: #FFABAB;">"ldsc.py --l2 --bfile {params.plink}  --ld-wind-cm 1 --out {params.odir}"</span>

</pre>
</div>

<p>
This script (<code>pull_ldscores_m50.R</code>) allows us to create subsets of the full baseline LD scores.  (see rule <code>cmp_baseline_ldscores2</code>)
</p>

<p>
The tangled file: <code>../scripts/pull_ldscores_m50.R</code> can be found here: <a href="https://github.com/ptb_workflowr/tree/master/scripts/pull_ldscores_m50.R">https://github.com/ptb_workflowr/tree/master/scripts/pull_ldscores_m50.R</a>
</p>
<div class="org-src-container">
<pre class="src src-R"><span style="color: #C04040;">library</span>(vroom)
<span style="color: #C04040;">library</span>(purrr)
<span style="color: #C04040;">library</span>(fs)

anno_bed_if <span style="color: #C04040;">&lt;-</span> snakemake@input[[<span style="color: #FFABAB;">"anno_bed"</span>]]
l2m_if <span style="color: #C04040;">&lt;-</span> snakemake@input[[<span style="color: #FFABAB;">"l2"</span>]]
l2m50_if <span style="color: #C04040;">&lt;-</span> snakemake@input[[<span style="color: #FFABAB;">"l2M_50"</span>]]


l2m_of <span style="color: #C04040;">&lt;-</span> snakemake@output[[<span style="color: #FFABAB;">"l2"</span>]]
l2m50_of <span style="color: #C04040;">&lt;-</span> snakemake@output[[<span style="color: #FFABAB;">"l2M_50"</span>]]
pull_features <span style="color: #C04040;">&lt;-</span> snakemake@params[[<span style="color: #FFABAB;">"features"</span>]]
anno_cols <span style="color: #C04040;">&lt;-</span> scan(anno_bed_if,what = character(),nlines = 1)[-c(1:4)]
keep_cols <span style="color: #C04040;">&lt;-</span> anno_cols <span style="color: #C04040;">%in%</span> pull_features


l2md <span style="color: #C04040;">&lt;-</span> scan(l2m_if,what=character(),nlines = 1)[keep_cols]
l2m50d <span style="color: #C04040;">&lt;-</span> scan(l2m50_if,what=character(),nlines = 1)[keep_cols]

write(paste0(l2md,collapse = <span style="color: #FFABAB;">"\t"</span>), l2m_of)
write(paste0(l2m50d,collapse = <span style="color: #FFABAB;">"\t"</span>),l2m50_of)
</pre>
</div>

<p>
This script (<code>pull_ldscores.R</code>) allows us to create subsets of the full baseline LD scores.  (see rule <code>cmp_baseline_ldscores</code>)
</p>

<p>
The tangled file: <code>../scripts/pull_ldscores.R</code> can be found here: <a href="https://github.com/ptb_workflowr/tree/master/scripts/pull_ldscores.R">https://github.com/ptb_workflowr/tree/master/scripts/pull_ldscores.R</a>
</p>
<div class="org-src-container">
<pre class="src src-R">
<span style="color: #C04040;">library</span>(vroom)
<span style="color: #C04040;">library</span>(purrr)
<span style="color: #C04040;">library</span>(fs)

anno_bed_if <span style="color: #C04040;">&lt;-</span> snakemake@input[[<span style="color: #FFABAB;">"anno_bed"</span>]]
l2gz_if <span style="color: #C04040;">&lt;-</span> snakemake@input[[<span style="color: #FFABAB;">"l2gz"</span>]]

anno_bed_of <span style="color: #C04040;">&lt;-</span> snakemake@output[[<span style="color: #FFABAB;">"anno_bed"</span>]]
l2gz_of <span style="color: #C04040;">&lt;-</span> snakemake@output[[<span style="color: #FFABAB;">"l2gz"</span>]]
pull_features <span style="color: #C04040;">&lt;-</span> snakemake@params[[<span style="color: #FFABAB;">"features"</span>]]

annot_prefix <span style="color: #C04040;">&lt;-</span> c(<span style="color: #FFABAB;">"CHR"</span>,<span style="color: #FFABAB;">"BP"</span>,<span style="color: #FFABAB;">"SNP"</span>,<span style="color: #FFABAB;">"CM"</span>)
annot_cols <span style="color: #C04040;">&lt;-</span> c(annot_prefix, pull_features)
names(annot_cols) <span style="color: #C04040;">&lt;-</span> annot_cols
annot_cols <span style="color: #C04040;">&lt;-</span> map(annot_cols, ~col_guess())
annot_cols <span style="color: #C04040;">&lt;-</span> vroom::cols_only(!!!annot_cols)

l2_prefix <span style="color: #C04040;">&lt;-</span> c(<span style="color: #FFABAB;">"CHR"</span>,<span style="color: #FFABAB;">"SNP"</span>,<span style="color: #FFABAB;">"BP"</span>)
l2_cols <span style="color: #C04040;">&lt;-</span> c(l2_prefix, paste0(pull_features, <span style="color: #FFABAB;">"L2"</span>))
names(l2_cols) <span style="color: #C04040;">&lt;-</span> l2_cols
l2_cols <span style="color: #C04040;">&lt;-</span> map(l2_cols, ~col_guess())
l2_cols <span style="color: #C04040;">&lt;-</span> vroom::cols_only(!!!l2_cols)

vroom::vroom_write(vroom::vroom(l2gz_if, delim = <span style="color: #FFABAB;">"\t"</span>,col_types = l2_cols),l2gz_of,delim = <span style="color: #FFABAB;">"\t"</span>)
vroom::vroom_write(vroom::vroom(anno_bed_if,delim = <span style="color: #FFABAB;">"\t"</span>,col_types = annot_cols),anno_bed_of,delim = <span style="color: #FFABAB;">"\t"</span>)

</pre>
</div>
</div>
</div>

<div id="outline-container-org47333f8" class="outline-3">
<h3 id="org47333f8">Computing LD scores and estimating heritability</h3>
<div class="outline-text-3" id="text-org47333f8">
<p>
These rules are where the action happens.  Stratified LD scores are computed (or subset) from baseline
annotations (or LD scores), and LD score regression is run.  There's an additional rule for estimating the co-heritability.
</p>

<p>
The tangled file: <code>../workflow/ldsc_snakefile</code> can be found here: <a href="https://github.com/ptb_workflowr/tree/master/workflow/ldsc_snakefile">https://github.com/ptb_workflowr/tree/master/workflow/ldsc_snakefile</a>
</p>
<div class="org-src-container">
<pre class="src src-snakemake">
<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">cmp_baseline_ldscores</span>:
    <span style="color: #99D6FF;">input</span>:
        <span style="color: #FF9326;">anno_bed</span>=config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"baseline/baselineLD.{chrom}.annot.gz"</span>,
        <span style="color: #FF9326;">l2gz</span>=config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"baseline/baselineLD.{chrom}.l2.ldscore.gz"</span>,
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">anno_bed</span>=config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"new_baseline/{new_base}.{chrom}.annot.gz"</span>,
        <span style="color: #FF9326;">l2gz</span>=config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"new_baseline/{new_base}.{chrom}.l2.ldscore.gz"</span>,
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">features</span>=<span style="color: #C8FF03;">lambda</span> <span style="color: #B46DCC;">wildcards</span>: all_annot.get(<span style="color: #B46DCC;">wildcards</span>.new_base),
        <span style="color: #FF9326;">anno</span>=<span style="color: #FFABAB;">"{new_base}"</span>
    <span style="color: #99D6FF;">script</span>:
        <span style="color: #FFABAB;">"../scripts/pull_ldscores.R"</span>

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">cmp_baseline_ldscores2</span>:
    <span style="color: #99D6FF;">input</span>:
        <span style="color: #FF9326;">anno_bed</span>=config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"baseline/baselineLD.{chrom}.annot.gz"</span>,
        <span style="color: #FF9326;">l2</span>=config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"baseline/baselineLD.{chrom}.l2.M"</span>,
        <span style="color: #FF9326;">l2M_50</span>=config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"baseline/baselineLD.{chrom}.l2.M_5_50"</span>
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">l2</span>=config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"new_baseline/{new_base}.{chrom}.l2.M"</span>,
        <span style="color: #FF9326;">l2M_50</span>=config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"new_baseline/{new_base}.{chrom}.l2.M_5_50"</span>
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">features</span>=<span style="color: #C8FF03;">lambda</span> <span style="color: #B46DCC;">wildcards</span>: all_annot.get(<span style="color: #B46DCC;">wildcards</span>.new_base),
        <span style="color: #FF9326;">anno</span>=<span style="color: #FFABAB;">"{new_base}"</span>
    <span style="color: #99D6FF;">script</span>:
        <span style="color: #FFABAB;">"../scripts/pull_ldscores_m50.R"</span>

<span style="color: #C8FF03;">def</span> <span style="color: #FFE203;">get_annot_files</span>(<span style="color: #B46DCC;">wildcards</span>):
        <span style="color: #C8FF03;">return</span> {<span style="color: #FFABAB;">'anno_l2'</span>:<span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"{anno_name}.{chrom}.l2.ldscore.gz"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23),anno_name=all_annot[<span style="color: #FFABAB;">'ptb_ldsc_model'</span>][<span style="color: #B46DCC;">wildcards</span>.anno_name]),
                <span style="color: #FFABAB;">'annotf'</span>:<span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"{anno_name}.{chrom}.annot.gz"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23),anno_name=all_annot[<span style="color: #FFABAB;">'ptb_ldsc_model'</span>][<span style="color: #B46DCC;">wildcards</span>.anno_name]),
                <span style="color: #FFABAB;">'baseline_l2'</span>:<span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"baseline/baselineLD.{chrom}.l2.ldscore.gz"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23)),
                <span style="color: #FFABAB;">'gwasf'</span>:config_d[<span style="color: #FFABAB;">'GWAS'</span>] +f<span style="color: #FFABAB;">"ldsc_input/{wildcards.gwas}_gwas.sumstats.gz"</span>,
                <span style="color: #FFABAB;">'baselinef'</span>:  <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'WEIGHTS'</span>] +<span style="color: #FFABAB;">"weights.hm3_noMHC.{chrom}.l2.ldscore.gz"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23)),
                <span style="color: #FFABAB;">'freqf'</span>:  <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'FRQF'</span>] +<span style="color: #FFABAB;">"1000G.EUR.QC.{chrom}.frq"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23)),
        }

<span style="color: #C8FF03;">def</span> <span style="color: #FFE203;">get_annot_pairs</span>(<span style="color: #B46DCC;">wildcards</span>):
        <span style="color: #C8FF03;">return</span> {<span style="color: #FFABAB;">'anno_l2'</span>:<span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"{anno_name}.{chrom}.l2.ldscore.gz"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23),anno_name=all_annot[<span style="color: #FFABAB;">'ptb_ldsc_model'</span>][<span style="color: #B46DCC;">wildcards</span>.anno_name]),
                <span style="color: #FFABAB;">'baseline_l2'</span>:<span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"baseline/baselineLD.{chrom}.l2.ldscore.gz"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23)),

                <span style="color: #FFABAB;">'gwasfA'</span>:config_d[<span style="color: #FFABAB;">'GWAS'</span>] +f<span style="color: #FFABAB;">"ldsc_input/{wildcards.gwasA}_gwas.sumstats.gz"</span>,
                <span style="color: #FFABAB;">'gwasfB'</span>:config_d[<span style="color: #FFABAB;">'GWAS'</span>] +f<span style="color: #FFABAB;">"ldsc_input/{wildcards.gwasB}_gwas.sumstats.gz"</span>,
                <span style="color: #FFABAB;">'baselinef'</span>:  <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'WEIGHTS'</span>] +<span style="color: #FFABAB;">"weights.hm3_noMHC.{chrom}.l2.ldscore.gz"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23)),
                <span style="color: #FFABAB;">'freqf'</span>:  <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'FRQF'</span>] +<span style="color: #FFABAB;">"1000G.EUR.QC.{chrom}.frq"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23)),
        }

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">run_ldsc</span>:
    <span style="color: #99D6FF;">input</span>:
        <span style="color: #B46DCC;">unpack</span>(get_annot_files)
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">dataf</span>=<span style="color: #FFABAB;">"results/{gwas}/{anno_name}.results"</span>
    <span style="color: #99D6FF;">log</span>:
        <span style="color: #FF9326;">tempf</span>=<span style="color: #B46DCC;">temp</span>(<span style="color: #FFABAB;">"{gwas}_{anno_name}.log"</span>)
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">annot</span>=<span style="color: #C8FF03;">lambda</span> <span style="color: #B46DCC;">wildcards</span>: <span style="color: #FFABAB;">','</span>.join(<span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"{anno_name}."</span>,anno_name=all_annot[<span style="color: #FFABAB;">'ptb_ldsc_model'</span>][<span style="color: #B46DCC;">wildcards</span>.anno_name])),
        <span style="color: #FF9326;">baseline</span>=config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"baseline/baselineLD."</span>,
        <span style="color: #FF9326;">weights</span>=config_d[<span style="color: #FFABAB;">'WEIGHTS'</span>]+<span style="color: #FFABAB;">"weights.hm3_noMHC."</span>,
        <span style="color: #FF9326;">frq</span>=config_d[<span style="color: #FFABAB;">'FRQF'</span>] +<span style="color: #FFABAB;">"1000G.EUR.QC."</span>,
        <span style="color: #FF9326;">odir</span>=<span style="color: #FFABAB;">"results/{gwas}/{anno_name}"</span>
    <span style="color: #99D6FF;">conda</span>:
        config_e[<span style="color: #FFABAB;">'ldsc'</span>]
    <span style="color: #99D6FF;">shell</span>:
        config_d[<span style="color: #FFABAB;">'LDSC'</span>]+<span style="color: #FFABAB;">"ldsc.py --h2 {input.gwasf} --ref-ld-chr {params.annot},{params.baseline} --w-ld-chr {params.weights} --thin-annot --overlap-annot --frqfile-chr {params.frq} --out {params.odir} "</span>

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">run_ldsc_cor</span>:
    <span style="color: #99D6FF;">input</span>:
        <span style="color: #B46DCC;">unpack</span>(get_annot_pairs)
    <span style="color: #99D6FF;">output</span>:
        <span style="color: #FF9326;">dataf</span>=<span style="color: #FFABAB;">"{gwasA},{gwasB}/{anno_name}.log"</span>
    <span style="color: #99D6FF;">params</span>:
        <span style="color: #FF9326;">annot</span>=<span style="color: #C8FF03;">lambda</span> <span style="color: #B46DCC;">wildcards</span>: <span style="color: #FFABAB;">','</span>.join(<span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"{anno_name}."</span>,anno_name=all_annot[<span style="color: #FFABAB;">'ptb_ldsc_model'</span>][<span style="color: #B46DCC;">wildcards</span>.anno_name])),
        <span style="color: #FF9326;">baseline</span>=config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"baseline/baselineLD."</span>,
        <span style="color: #FF9326;">weights</span>=config_d[<span style="color: #FFABAB;">'WEIGHTS'</span>]+<span style="color: #FFABAB;">"weights.hm3_noMHC."</span>,
        <span style="color: #FF9326;">frq</span>=config_d[<span style="color: #FFABAB;">'FRQF'</span>] +<span style="color: #FFABAB;">"1000G.EUR.QC."</span>,
        <span style="color: #FF9326;">odir</span>=<span style="color: #FFABAB;">"{gwasA},{gwasB}/{anno_name}"</span>
    <span style="color: #99D6FF;">conda</span>:
        config_e[<span style="color: #FFABAB;">'ldsc'</span>]
    <span style="color: #99D6FF;">shell</span>:
        config_d[<span style="color: #FFABAB;">'LDSC'</span>]+<span style="color: #FFABAB;">"ldsc.py --rg {input.gwasfA},{input.gwasfB} --ref-ld-chr {params.annot},{params.baseline} --w-ld-chr {params.weights} --thin-annot --overlap-annot --frqfile-chr {params.frq} --out {params.odir} "</span>

<span style="color: #C8FF03;">def</span> <span style="color: #FFE203;">get_new_annot_files</span>(<span style="color: #B46DCC;">wildcards</span>):

        <span style="color: #C8FF03;">return</span> {<span style="color: #FFABAB;">'anno_l2'</span>:<span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"{anno_name}.{chrom}.l2.ldscore.gz"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23),anno_name=all_annot[<span style="color: #FFABAB;">'ptb_ldsc_model'</span>][<span style="color: #B46DCC;">wildcards</span>.anno_name]),
                <span style="color: #FFABAB;">'baseline_l2'</span>:<span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"new_baseline/{new_base}.{chrom}.l2.ldscore.gz"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23),new_base = <span style="color: #B46DCC;">wildcards</span>.new_base),
                <span style="color: #FFABAB;">'baseline_l2m'</span>:<span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"new_baseline/{new_base}.{chrom}.l2.M"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23),new_base = <span style="color: #B46DCC;">wildcards</span>.new_base),
                <span style="color: #FFABAB;">'baseline_l2m50'</span>:<span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>] +<span style="color: #FFABAB;">"new_baseline/{new_base}.{chrom}.l2.M_5_50"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23),new_base = <span style="color: #B46DCC;">wildcards</span>.new_base),

                <span style="color: #FFABAB;">'baselinef'</span>:  <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'WEIGHTS'</span>] +<span style="color: #FFABAB;">"weights.hm3_noMHC.{chrom}.l2.ldscore.gz"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23)),
                <span style="color: #FFABAB;">'freqf'</span>:  <span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'FRQF'</span>] +<span style="color: #FFABAB;">"1000G.EUR.QC.{chrom}.frq"</span>,chrom=<span style="color: #B46DCC;">range</span>(1,23)),
        }

<span style="color: #C8FF03;">rule</span> <span style="color: #FFE203;">run_new_ldsc</span>:
      <span style="color: #99D6FF;">input</span>:
          <span style="color: #B46DCC;">unpack</span>(get_new_annot_files)
      <span style="color: #99D6FF;">output</span>:
            <span style="color: #FF9326;">dataf</span>=<span style="color: #FFABAB;">"results/{gwas}/{new_base}_{anno_name}.results"</span>
      <span style="color: #99D6FF;">log</span>:
          <span style="color: #FF9326;">tempf</span>=<span style="color: #B46DCC;">temp</span>(<span style="color: #FFABAB;">"{gwas}/{new_base}_{anno_name}.log"</span>)
      <span style="color: #99D6FF;">params</span>:
          <span style="color: #FF9326;">annot</span>=<span style="color: #C8FF03;">lambda</span> <span style="color: #B46DCC;">wildcards</span>: <span style="color: #FFABAB;">','</span>.join(<span style="color: #B46DCC;">expand</span>(config_d[<span style="color: #FFABAB;">'L2'</span>]+<span style="color: #FFABAB;">"{anno_name}."</span>,anno_name=all_annot[<span style="color: #FFABAB;">'ptb_ldsc_model'</span>][<span style="color: #B46DCC;">wildcards</span>.anno_name])),
          <span style="color: #FF9326;">baseline</span>=config_d[<span style="color: #FFABAB;">"L2"</span>]+<span style="color: #FFABAB;">"new_baseline/{new_base}."</span>,
          <span style="color: #FF9326;">weights</span>=config_d[<span style="color: #FFABAB;">'WEIGHTS'</span>]+<span style="color: #FFABAB;">"weights.hm3_noMHC."</span>,
          <span style="color: #FF9326;">frq</span>=config_d[<span style="color: #FFABAB;">'FRQF'</span>] +<span style="color: #FFABAB;">"1000G.EUR.QC."</span>,
          <span style="color: #FF9326;">odir</span>=<span style="color: #FFABAB;">"results/{gwas}/{new_base}_{anno_name}"</span>
      <span style="color: #99D6FF;">conda</span>:
          config_e[<span style="color: #FFABAB;">'ldsc'</span>]
      <span style="color: #99D6FF;">shell</span>:
          config_d[<span style="color: #FFABAB;">'LDSC'</span>]+<span style="color: #FFABAB;">"ldsc.py --h2 {input.gwasf} --ref-ld-chr {params.annot},{params.baseline} --w-ld-chr {params.weights} --thin-annot --overlap-annot --frqfile-chr {params.frq} --out {params.odir}"</span>


</pre>
</div>
</div>
</div>
</div>
</div>
<div id="postamble" class="status">
<p class="author">Author: Nicholas Knoblauch</p>
<p class="date">Created: 2020-05-08 Fri 11:22</p>
<p class="validation"><a href="http://validator.w3.org/check?uri=referer">Validate</a></p>
</div>
</body>
</html>
