<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="CreRecombinase" />

<meta name="date" content="2019-07-18" />

<title>Fine Mapping using Torus and Susie</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.7/datatables.js"></script>
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ptb</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Fine Mapping using Torus and Susie</h1>
<h4 class="author">CreRecombinase</h4>
<h4 class="date">2019-07-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-07-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ptb_workflowr/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190313code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190313)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190313code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190313)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/home/nwknoblauch/Dropbox/Repos/ptb_workflowr/cache/rcc2/
</td>
<td style="text-align:left;">
cache/rcc2
</td>
</tr>
<tr>
<td style="text-align:left;">
~/Dropbox/Repos/ptb_workflowr/R/config.R
</td>
<td style="text-align:left;">
R/config.R
</td>
</tr>
<tr>
<td style="text-align:left;">
~/Dropbox/Repos/ptb_workflowr/R/plan.R
</td>
<td style="text-align:left;">
R/plan.R
</td>
</tr>
<tr>
<td style="text-align:left;">
~/Dropbox/Repos/ptb_workflowr/code/snp_gene.R
</td>
<td style="text-align:left;">
code/snp_gene.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomCreRecombinaseptbworkflowrtree1fbd59fcf51ea34d601d8c630d0c54ca7cc0b04dtargetblank1fbd59fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/CreRecombinase/ptb_workflowr/tree/1fbd59fcf51ea34d601d8c630d0c54ca7cc0b04d" target="_blank">1fbd59f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomCreRecombinaseptbworkflowrtree1fbd59fcf51ea34d601d8c630d0c54ca7cc0b04dtargetblank1fbd59fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:     (dellxps's conflicted copy 2019-07-30).Rhistory
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .drake/
    Ignored:    analysis/PTB_features (dellxps's conflicted copy 2019-06-19).Rmd
    Ignored:    analysis/figure/
    Ignored:    cache/
    Ignored:    data/NA_character_.RDS
    Ignored:    data/ag_df.RDS
    Ignored:    data/annotations.h5
    Ignored:    data/big_hic_t.tsv.gz
    Ignored:    data/chr19_snpdata_hm3only.RDS
    Ignored:    data/eqtl_gwas_file.tsv.gz
    Ignored:    data/evd_list_chr19_hm3.RDS
    Ignored:    data/exons.bed
    Ignored:    data/fat_eqtl.tsv.gz
    Ignored:    data/gencode.v28.annotation.gff3.gz
    Ignored:    data/genes.bed
    Ignored:    data/gwas_file.tsv.gz
    Ignored:    data/gwas_ptb_file.tsv.gz
    Ignored:    data/gwas_ptb_file_i.tsv.zstd
    Ignored:    data/gwas_scz_file.tsv.gz
    Ignored:    data/hic_t.tsv.gz
    Ignored:    data/little_gwas_i.tsv.zstd
    Ignored:    data/matched_snp.RDS
    Ignored:    data/new_cache_log.RDS
    Ignored:    data/ngwas_df.h5
    Ignored:    data/old_cache_log.RDS
    Ignored:    data/retd.RDS
    Ignored:    data/shuffeqtl_gwas_file.tsv.gz
    Ignored:    data/split_d.h5
    Ignored:    data/sub_gwas_ptb.tsv.gz
    Ignored:    data/susie_r.RDS
    Ignored:    data/ut_eqtl.tsv.gz
    Ignored:    old_index/

Untracked files:
    Untracked:  .drake_history/
    Untracked:  analysis/23_and_me.Rmd
    Untracked:  analysis/LD.Rmd
    Untracked:  analysis/awd.RData
    Untracked:  analysis/dataset_SNPs_mismatches-reference.txt
    Untracked:  analysis/enrichment_analysis.Rmd
    Untracked:  analysis/ldshrink_drake.Rmd
    Untracked:  analysis/mws.RData
    Untracked:  analysis/test_knit_code.Rmd
    Untracked:  analysis/test_knit_code2.Rmd
    Untracked:  code/
    Untracked:  config.yml
    Untracked:  config/four_features.yaml
    Untracked:  config/four_features_repro_ctrl.yaml
    Untracked:  config/four_features_repro_dec.yaml
    Untracked:  extract_results_main.R
    Untracked:  extract_sigma_g.R
    Untracked:  false_positives.dsc
    Untracked:  power.dsc
    Untracked:  ptb_cache/
    Untracked:  sigma_g.dsc
    Untracked:  sim_results/
    Untracked:  sub_drake.sh
    Untracked:  summarize_sigma_g.R

Unstaged changes:
    Modified:   .Rprofile
    Modified:   R/functions.R
    Modified:   R/plan.R
    Modified:   analysis/PTB_features.Rmd
    Modified:   analysis/multifeature.Rmd
    Modified:   analysis/multifeature_update.Rmd
    Modified:   config/workflow_params_desktop.yaml
    Modified:   config/workflow_params_xps.yaml
    Modified:   make.R
    Modified:   slurm_clustermq.tmpl
    Modified:   torque_clustermq.tmpl

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/1fbd59fcf51ea34d601d8c630d0c54ca7cc0b04d/analysis/marginal_joint.Rmd" target="_blank">1fbd59f</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
<td>
workflowr::wflow_publish(“analysis/marginal_joint.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/marginal_joint.html" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/7f7ddfdf5112990fb289444f9f66d38223b83549/analysis/marginal_joint.Rmd" target="_blank">7f7ddfd</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
<td>
workflowr::wflow_publish(“analysis/marginal_joint.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/206221daffbe870987637b30fe0d38ca3e616588/docs/marginal_joint.html" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
<td>
Build site.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In this document I will summarize my current progress analyzing the latest the GWAS results. I’ll explain briefly how <code>torus</code> does enrichment analysis (and what I mean by “enrichment analysis”). Then I’ll go over the results of that analysis. After that I’ll give a brief overview of susie, and then I’ll show some results. Before any of that, I’m going to talk a little bit about the dataset(s) I’m working with.</p>
<div id="the-gwas" class="section level2">
<h2>The GWAS</h2>
<div class="figure">
<img src="../data/orig_gwas.png" alt="" />
<p class="caption">manhattan</p>
</div>
<p>A few numbers about the GWAS summary statistics:</p>
<ul>
<li><code>14991824</code> variants</li>
<li>top hit has <span class="math inline">\(p\)</span>=3.03610^{-26}</li>
<li><code>p&lt;1e-8</code> on chromosomes 1,3,5</li>
</ul>
<pre class="r"><code>ggplot(gwas_df_ptb, aes(x=N)) + geom_histogram(bins=50) +
  xlab(&quot;Sample size&quot;)</code></pre>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-2-1.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># eff_ld_df &lt;- dplyr::semi_join(gwas_df_ptb,select(top_gwas_reg,region_id)) %&gt;% group_by(region_id) %&gt;% summarise(chr=paste0(&quot;chr&quot;,chrom[1]),start=min(pos),end=max(pos)+1L) %&gt;% GenomicRanges::makeGRangesFromDataFrame()
# 
# ldr &lt;- eff_ld_df[1]
# library(ggbio)
# ideo &lt;- ggbio::Ideogram(genome=&quot;hg19&quot;)
# ideo+xlim(eff_ld_df[1])
# data(genesymbol, package = &quot;biovizBase&quot;)
# 
# 
# 
# gsi &lt;- join_overlap_intersect_within(genesymbol,eff_ld_df)
# gsi_df &lt;- join_overlap_intersect_within(genesymbol,eff_ld_df[1])
# p.txdb &lt;- autoplot(Homo.sapiens, which = gsi_df)</code></pre>
<p>#Enrichment analysis and fine mapping pipeline overview</p>
<ol style="list-style-type: decimal">
<li>Estimate the relationship between a particular genomic annotation (e.g ATAC-seq peaks) and GWAS significance genome-wide (Torus)</li>
<li>Use the enrichment estimate to specify a per-variant prior.</li>
<li>Using the prior and a reference LD panel, identify putative causal variants (<code>susie</code>)</li>
</ol>
<p>For both the enrichment analysis and the fine mapping, the genome is broken in to chunks of size 50000. For torus (enrichment) there is an assumption that there is at most 1 causal variant per chunk, and for susie (fine-mapping) the assumption is that there are at most 10. These assumptions make chunk size an important parameter when performing either fine mapping or enrichment analysis.</p>
</div>
<div id="genomics-data" class="section level2">
<h2>Genomics Data</h2>
<p>Noboru has provided me with <code>bed</code> file annotations of the genome generated from the various experiments performed on the cell lines.</p>
<ol style="list-style-type: decimal">
<li>ATAC seq</li>
<li>H3K4me1 chip</li>
<li>H3K4me3 chip</li>
<li>H3K27ac chip</li>
</ol>
<p>Within these categories there are subcategories:</p>
<ol style="list-style-type: decimal">
<li>Per-cell-line. Each cell line was either a control (<code>ctr</code>) or underwent decidualization <code>dec</code>.</li>
<li>Annotations consistent across the three control or decidualized cell line samples.</li>
<li>Differental peaks. Peaks that have increased read counts in decidualized over control</li>
</ol>
<p>In addition I have:</p>
<ul>
<li>Endometrial eQTL (Ober Lab)</li>
<li>Hi-C from one cell line <code>hic_all_interacting_DT1_dTL4_D_48h</code></li>
<li>Annotation predicting repressive regions (from Hoffman et al.)</li>
</ul>
</div>
<div id="all-the-univariate-results" class="section level2">
<h2>All the univariate results</h2>
<pre class="r"><code>filter(irl,term !=&quot;Intercept&quot;) %&gt;%  ggplot(aes(x=estimate,y=term))+geom_point()+geom_errorbarh(aes(xmin=low,xmax=high))+geom_vline(xintercept=0)+ggtitle(&quot;All Univariate Enrichment Estimates&quot;)</code></pre>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-4-1.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="multivariate-effect-size-estimates" class="section level2">
<h2>Multivariate effect size estimates</h2>
<p>Below are the multivariate effect size estimates for 5 features that came out of a forward selection.</p>
<pre class="r"><code>dplyr::select(mrl,-lik,-z,-sd) %&gt;%  
  DT::datatable()</code></pre>
<div id="htmlwidget-f22c17590d6b3f85a2bb" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f22c17590d6b3f85a2bb">{"x":{"filter":"none","data":[["1","2","3","4","5","6"],["Intercept","atac-seq-pooled-DSC1-dec-ATAC","chip-seq-dec_up-H3K27ac","chip-seq-dec_up-H3K4me1","chip-seq-pooled-DSC2-ctr-H3K4me1","eQTL_0.05_FDR"],[-11.418,-3.951,2.452,-2.29,2.882,6.306],[-11.444,-6.083,1.121,-3.53,2.011,0.617],[-11.392,-1.819,3.782,-1.051,3.754,11.995],[0,0.000280959192802085,0.000305302456283101,0.000294970507614499,8.85394184348712e-11,0.0298125730935272]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>term<\/th>\n      <th>estimate<\/th>\n      <th>low<\/th>\n      <th>high<\/th>\n      <th>p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code> mrl %&gt;% 
  filter(term!=&quot;Intercept&quot;) %&gt;% 
  ggplot(aes(x=estimate,y=term)) + 
  geom_point() + 
  geom_errorbarh(aes(xmin=low,xmax=high)) + 
  geom_vline(xintercept=0) + 
  ggtitle(&quot;Multivariate Enrichment Estimates&quot;)</code></pre>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-6-1.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code> amrl%&gt;% 
  filter(term!=&quot;Intercept&quot;) %&gt;% 
  ggplot(aes(x=estimate,y=term)) + 
  geom_point() + 
  geom_errorbarh(aes(xmin=low,xmax=high)) + 
  geom_vline(xintercept=0) + 
  ggtitle(&quot;Multivariate Enrichment Estimates (of alternative model)&quot;)</code></pre>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-7-1.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>It should be noted that not all of these features are significant when fit as 5 univariate models.</p>
<pre class="r"><code> irl %&gt;% filter(term!=&quot;Intercept&quot;,term %in% best_terms) %&gt;%  ggplot(aes(x=estimate,y=term))+geom_point()+geom_errorbarh(aes(xmin=low,xmax=high))+geom_vline(xintercept=0)+ggtitle(&quot;Univariate Enrichment Estimates&quot;)</code></pre>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-8-1.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>brl &lt;-  irl %&gt;%  filter(term!=&quot;Intercept&quot;) %&gt;% 
  select(-lik,-sd,-z,-p) %&gt;% 
  inner_join(select(mrl,term,estimate,low,high),by = c(&quot;term&quot;),suffix = c(&quot;_univariate&quot;,&quot;_multivariate&quot;))

abrl &lt;-  irl%&gt;%  filter(term!=&quot;Intercept&quot;) %&gt;% 
  select(-lik,-sd,-z,-p) %&gt;% 
  inner_join(select(amrl,term,estimate,low,high),by = c(&quot;term&quot;),suffix = c(&quot;_univariate&quot;,&quot;_multivariate&quot;))

ggplot(brl,aes(x=estimate_univariate,y=estimate_multivariate,col=term))+geom_errorbarh(aes(xmin=low_univariate,xmax=high_univariate))+geom_errorbar(aes(ymin=low_multivariate,ymax=high_multivariate))+geom_point()+geom_hline(yintercept=0)+geom_vline(xintercept=0)+ggtitle(&quot;Univariate vs Multivariate effect size estimates&quot;)</code></pre>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-9-1.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(abrl,aes(x=estimate_univariate,y=estimate_multivariate,col=term))+geom_errorbarh(aes(xmin=low_univariate,xmax=high_univariate))+geom_errorbar(aes(ymin=low_multivariate,ymax=high_multivariate))+geom_point()+geom_hline(yintercept=0)+geom_vline(xintercept=0)+ggtitle(&quot;Univariate vs Multivariate effect size estimates (of alternate model)&quot;)</code></pre>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-10-1.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="fine-mapping-results" class="section level1">
<h1>Fine mapping results</h1>
<p>Each SNP that underwent fine mapping has a <span class="math inline">\(p\)</span>-value, a prior, and a posterior inclusion probability, or pip, which is the predicted probability that the SNP is a causal variant (i.e that the effect size estimate is not a sample from the null distribution).</p>
<p>Below is an example of what this looks like for a region</p>
<pre class="r"><code>af_obj &lt;- cc$list()[str_detect(cc$list(),&quot;^af_susie&quot;)]
af_susie_res &lt;- map_df(af_obj,~readd(.x,character_only = T,cache=cc))




f_obj &lt;- cc$list()[str_detect(cc$list(),&quot;^f_susie&quot;)]
all_susie_res &lt;- map_df(f_obj,~readd(.x,character_only = T,cache=cc))

tsr &lt;- select(top_gwas_reg,region_id,pz)</code></pre>
<pre class="r"><code>idfr &lt;- function(res_df,tsrdf,idx=1L){
  tt_susie_res &lt;- semi_join(res_df, slice(tsrdf,idx)) %&gt;%
  dplyr::mutate(log10_p=-log10(p))

  Rd &lt;- cor(geno_d$genotypes[,tt_susie_res$ld_id])
  top_tt_id &lt;- mutate(tt_susie_res,tid=1:n()) %&gt;% 
  dplyr::filter(p==min(p)) %&gt;% 
  dplyr::pull(tid)
  stopifnot(length(top_tt_id)==1)
  tt_susie_res &lt;- mutate(tt_susie_res,tr=Rd[,top_tt_id])

tt_susie_res %&gt;% 
    dplyr::select(chrom,pos,log10_p,prior,pip,tr) %&gt;% tidyr::gather(key=&quot;feature&quot;,value=&quot;value&quot;,log10_p,prior,pip)
}

susie_ires &lt;- map(1:nrow(tsr),~idfr(all_susie_res,tsr,.x))</code></pre>
<pre><code>Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;</code></pre>
<pre class="r"><code>asusie_ires &lt;- map(1:nrow(tsr),~idfr(af_susie_res,tsr,.x))</code></pre>
<pre><code>Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;
Joining, by = &quot;region_id&quot;</code></pre>
<div id="susie-plots-of-top-regions" class="section level2">
<h2>Susie plots of top regions</h2>
<pre class="r"><code>plot_sr &lt;- function(t_susie_res,text){
  tchr &lt;- c(t_susie_res$chrom[1],range(t_susie_res$pos))
  gb_df &lt;- gene_body_df(tchr[1],tchr[2],tchr[3],dbdir) 
  
  title &lt;- glue::glue(&quot;Chromosome {tchr[1]},{tchr[2]}-{tchr[3]} {text}&quot;)
  filter(t_susie_res)  %&gt;% 
    rename(r_2=tr) %&gt;% 
    mutate(r_2=r_2^2) %&gt;% 
    dplyr::filter(feature!=&quot;prior&quot;) %&gt;%  
    ggplot(aes(x=pos,y=value,col=r_2)) +
    geom_point() + 
    facet_grid(feature~.,scales = &quot;free_y&quot;) +
    ggtitle(title) +
    theme(axis.text.x = element_text(angle = 90)) +
    scale_x_continuous(label=unit_format(unit=&quot;Mb&quot;,scale=1e-6,accuracy = .01))
}

all_plots &lt;- map(susie_ires,plot_sr,text=&quot;Forward Selection Model&quot;)

walk(all_plots,print)</code></pre>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-1.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-2">
Past versions of unnamed-chunk-13-2.png
</button>
</p>
<div id="fig-unnamed-chunk-13-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-2.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-3">
Past versions of unnamed-chunk-13-3.png
</button>
</p>
<div id="fig-unnamed-chunk-13-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-3.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-4">
Past versions of unnamed-chunk-13-4.png
</button>
</p>
<div id="fig-unnamed-chunk-13-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-4.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-5">
Past versions of unnamed-chunk-13-5.png
</button>
</p>
<div id="fig-unnamed-chunk-13-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-5.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-6">
Past versions of unnamed-chunk-13-6.png
</button>
</p>
<div id="fig-unnamed-chunk-13-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-6.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-7">
Past versions of unnamed-chunk-13-7.png
</button>
</p>
<div id="fig-unnamed-chunk-13-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-7.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-8">
Past versions of unnamed-chunk-13-8.png
</button>
</p>
<div id="fig-unnamed-chunk-13-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-8.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-9">
Past versions of unnamed-chunk-13-9.png
</button>
</p>
<div id="fig-unnamed-chunk-13-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-9.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-10">
Past versions of unnamed-chunk-13-10.png
</button>
</p>
<div id="fig-unnamed-chunk-13-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-10.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-11">
Past versions of unnamed-chunk-13-11.png
</button>
</p>
<div id="fig-unnamed-chunk-13-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-11.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-12">
Past versions of unnamed-chunk-13-12.png
</button>
</p>
<div id="fig-unnamed-chunk-13-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-12.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-13">
Past versions of unnamed-chunk-13-13.png
</button>
</p>
<div id="fig-unnamed-chunk-13-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-13.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-14">
Past versions of unnamed-chunk-13-14.png
</button>
</p>
<div id="fig-unnamed-chunk-13-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-14.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-13-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-15">
Past versions of unnamed-chunk-13-15.png
</button>
</p>
<div id="fig-unnamed-chunk-13-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/206221daffbe870987637b30fe0d38ca3e616588/docs/figure/marginal_joint.Rmd/unnamed-chunk-13-15.png" target="_blank">206221d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="susie-plots-of-top-regions-alternate-model" class="section level2">
<h2>Susie plots of top regions (alternate model)</h2>
<pre class="r"><code>a_all_plots &lt;- map(asusie_ires,plot_sr,text=&quot;Three Parameter Model&quot;)
walk(a_all_plots,print)</code></pre>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-1.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-2.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-3">
Past versions of unnamed-chunk-14-3.png
</button>
</p>
<div id="fig-unnamed-chunk-14-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-3.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-4">
Past versions of unnamed-chunk-14-4.png
</button>
</p>
<div id="fig-unnamed-chunk-14-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-4.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-5">
Past versions of unnamed-chunk-14-5.png
</button>
</p>
<div id="fig-unnamed-chunk-14-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-5.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-6">
Past versions of unnamed-chunk-14-6.png
</button>
</p>
<div id="fig-unnamed-chunk-14-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-6.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-7">
Past versions of unnamed-chunk-14-7.png
</button>
</p>
<div id="fig-unnamed-chunk-14-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-7.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-8">
Past versions of unnamed-chunk-14-8.png
</button>
</p>
<div id="fig-unnamed-chunk-14-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-8.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-9">
Past versions of unnamed-chunk-14-9.png
</button>
</p>
<div id="fig-unnamed-chunk-14-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-9.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-10">
Past versions of unnamed-chunk-14-10.png
</button>
</p>
<div id="fig-unnamed-chunk-14-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-10.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-11">
Past versions of unnamed-chunk-14-11.png
</button>
</p>
<div id="fig-unnamed-chunk-14-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-11.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-12">
Past versions of unnamed-chunk-14-12.png
</button>
</p>
<div id="fig-unnamed-chunk-14-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-12.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-13">
Past versions of unnamed-chunk-14-13.png
</button>
</p>
<div id="fig-unnamed-chunk-14-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-13.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-14">
Past versions of unnamed-chunk-14-14.png
</button>
</p>
<div id="fig-unnamed-chunk-14-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-14.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/marginal_joint.Rmd/unnamed-chunk-14-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-15">
Past versions of unnamed-chunk-14-15.png
</button>
</p>
<div id="fig-unnamed-chunk-14-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/2755ec95b2edff9d4752e6923ff0ecd6c6d4ac75/docs/figure/marginal_joint.Rmd/unnamed-chunk-14-15.png" target="_blank">2755ec9</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-07-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Manjaro Linux

Matrix products: default
BLAS/LAPACK: /usr/lib/libopenblas_haswellp-r0.3.6.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tidyselect_0.2.5   RSSp_0.9.0.9000    ldmap_0.0.0.9000  
 [4] daprcpp_1.0.0.9000 ldshrink_1.0-1     bigsnpr_0.11.5    
 [7] bigstatsr_0.9.9    vroom_1.0.2.9000   RSQLite_2.1.1     
[10] glue_1.3.1         drake_7.4.0.9000   fs_1.3.1          
[13] susieR_0.8.1.0525  here_0.1           forcats_0.4.0     
[16] stringr_1.4.0      dplyr_0.8.3        purrr_0.3.2       
[19] readr_1.3.1        tidyr_0.8.3        tibble_2.1.3      
[22] ggplot2_3.2.0      tidyverse_1.2.1    dbplyr_1.4.2      
[25] MonetDBLite_0.6.1  gtable_0.3.0       gridExtra_2.3     
[28] scales_1.0.0      

loaded via a namespace (and not attached):
 [1] colorspace_1.4-1       RcppEigen_0.3.3.5.0    rprojroot_1.3-2       
 [4] XVector_0.24.0         GenomicRanges_1.36.0   rstudioapi_0.10       
 [7] DT_0.7                 bit64_0.9-7            lubridate_1.7.4       
[10] xml2_1.2.0             codetools_0.2-16       knitr_1.23            
[13] jsonlite_1.6           workflowr_1.4.0        broom_0.5.2           
[16] shiny_1.3.2            compiler_3.6.1         httr_1.4.0            
[19] backports_1.1.4        assertthat_0.2.1       Matrix_1.2-17         
[22] lazyeval_0.2.2         cli_1.1.0              later_0.8.0           
[25] htmltools_0.3.6        tools_3.6.1            igraph_1.2.4.1        
[28] GenomeInfoDbData_1.2.1 reshape2_1.4.3         Rcpp_1.0.2            
[31] cellranger_1.1.0       nlme_3.1-140           iterators_1.0.10      
[34] crosstalk_1.0.0        wavethresh_4.6.8       xfun_0.7              
[37] rvest_0.3.4            mime_0.7               zlibbioc_1.30.0       
[40] MASS_7.3-51.4          hms_0.4.2              promises_1.0.1        
[43] parallel_3.6.1         yaml_2.2.0             memoise_1.1.0         
[46] stringi_1.4.3          highr_0.8              S4Vectors_0.22.0      
[49] foreach_1.4.4          BiocGenerics_0.30.0    filelock_1.0.2        
[52] storr_1.2.2            GenomeInfoDb_1.20.0    rlang_0.4.0.9000      
[55] pkgconfig_2.0.2        bitops_1.0-6           evaluate_0.14         
[58] lattice_0.20-38        htmlwidgets_1.3        labeling_0.3          
[61] cowplot_1.0.0          bit_1.1-14             plyr_1.8.4            
[64] magrittr_1.5           R6_2.4.0               IRanges_2.18.1        
[67] generics_0.0.2         base64url_1.4          txtq_0.1.3            
[70] DBI_1.0.0              pillar_1.4.2           haven_2.1.0           
[73] whisker_0.3-2          withr_2.1.2            RCurl_1.95-4.12       
[76] modelr_0.1.4           crayon_1.3.4           rmarkdown_1.13        
[79] grid_3.6.1             readxl_1.3.1           data.table_1.12.2     
[82] blob_1.1.1             git2r_0.26.1           digest_0.6.20         
[85] xtable_1.8-4           httpuv_1.5.1           RcppParallel_4.4.3    
[88] stats4_3.6.1           munsell_0.5.0         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
