<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="CreRecombinase" />

<meta name="date" content="2019-09-04" />

<title>region_discovery</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.7.2/datatables.js"></script>
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ptb</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">region_discovery</h1>
<h4 class="author">CreRecombinase</h4>
<h4 class="date">2019-09-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-09-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>ptb_workflowr/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190313code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190313)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190313code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190313)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomCreRecombinaseptbworkflowrtree88e89ecae5f828c8389b81cb497e4dd273484933targetblank88e89eca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/CreRecombinase/ptb_workflowr/tree/88e89ecae5f828c8389b81cb497e4dd273484933" target="_blank">88e89ec</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomCreRecombinaseptbworkflowrtree88e89ecae5f828c8389b81cb497e4dd273484933targetblank88e89eca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    .drake/
    Ignored:    analysis/PTB_features (dellxps's conflicted copy 2019-06-19).Rmd
    Ignored:    cache/
    Ignored:    data/NA_character_.RDS
    Ignored:    data/ag_df.RDS
    Ignored:    data/annotations.h5
    Ignored:    data/big_hic_t.tsv.gz
    Ignored:    data/chr19_snpdata_hm3only.RDS
    Ignored:    data/endometrium.jpg
    Ignored:    data/eqtl_gwas_file.tsv.gz
    Ignored:    data/evd_list_chr19_hm3.RDS
    Ignored:    data/exons.bed
    Ignored:    data/fat_eqtl.tsv.gz
    Ignored:    data/gencode.v28.annotation.gff3.gz
    Ignored:    data/genes.bed
    Ignored:    data/gwas_file.tsv.gz
    Ignored:    data/gwas_ptb_file.tsv.gz
    Ignored:    data/gwas_ptb_file_i.tsv.zstd
    Ignored:    data/gwas_scz_file.tsv.gz
    Ignored:    data/hic_t.tsv.gz
    Ignored:    data/little_gwas_i.tsv.zstd
    Ignored:    data/matched_snp.RDS
    Ignored:    data/new_cache_log.RDS
    Ignored:    data/ngwas_df.h5
    Ignored:    data/old_cache_log.RDS
    Ignored:    data/ptb_cred.yml
    Ignored:    data/retd.RDS
    Ignored:    data/shuffeqtl_gwas_file.tsv.gz
    Ignored:    data/split_d.h5
    Ignored:    data/sub_gwas_ptb.tsv.gz
    Ignored:    data/susie_r.RDS
    Ignored:    data/ut_eqtl.tsv.gz
    Ignored:    old_index/

Untracked files:
    Untracked:  analysis/23_and_me.Rmd
    Untracked:  analysis/LD.Rmd
    Untracked:  analysis/awd.RData
    Untracked:  analysis/dataset_SNPs_mismatches-reference.txt
    Untracked:  analysis/enrichment_analysis.Rmd
    Untracked:  analysis/finemap_plot.Rmd
    Untracked:  analysis/joint_annotations.Rmd
    Untracked:  analysis/ldshrink_drake.Rmd
    Untracked:  analysis/mws.RData
    Untracked:  analysis/test_knit_code.Rmd
    Untracked:  analysis/test_knit_code2.Rmd
    Untracked:  bd.txt
    Untracked:  code/
    Untracked:  config.yml
    Untracked:  config/four_features.yaml
    Untracked:  config/four_features_repro_ctrl.yaml
    Untracked:  config/four_features_repro_dec.yaml
    Untracked:  config/gwas.yaml
    Untracked:  de.txt
    Untracked:  extract_results_main.R
    Untracked:  extract_sigma_g.R
    Untracked:  false_positives.dsc
    Untracked:  marginal_joint.R
    Untracked:  power.dsc
    Untracked:  presentations/
    Untracked:  ptb_cache/
    Untracked:  report.Rmd
    Untracked:  sigma_g.dsc
    Untracked:  sim_results/
    Untracked:  sub_drake.sh
    Untracked:  summarize_sigma_g.R
    Untracked:  tmp.txt.gz

Unstaged changes:
    Modified:   .Rprofile
    Modified:   R/functions.R
    Modified:   R/plan.R
    Modified:   config/data/desktop.yaml
    Modified:   config/features.yaml
    Modified:   make.R
    Deleted:    renv.lock
    Deleted:    renv/.gitignore
    Deleted:    renv/activate.R
    Deleted:    renv/settings.dcf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/88e89ecae5f828c8389b81cb497e4dd273484933/analysis/region_discovery.Rmd" target="_blank">88e89ec</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-09-17
</td>
<td>
workflowr::wflow_publish(“analysis/region_discovery.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/9c1ed19819bc485f2879a28ab82aa1c6c3920aa7/docs/region_discovery.html" target="_blank">9c1ed19</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/58d859e6d86e0ad7ffcf8bcb8ff754c680329031/analysis/region_discovery.Rmd" target="_blank">58d859e</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-09-16
</td>
<td>
workflowr::wflow_publish(“analysis/region_discovery.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/039f99addd13b416d09682ea50499c88c12ef476/docs/region_discovery.html" target="_blank">039f99a</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/bd459ffae2b0d07178364f23fc8e6cd1fdca5939/docs/region_discovery.html" target="_blank">bd459ff</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/16fe586b347338ef8e07bdac00b6d14698004d03/analysis/region_discovery.Rmd" target="_blank">16fe586</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-09-16
</td>
<td>
workflowr::wflow_publish(“analysis/region_discovery.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/5b1d6dac6576a22c12fe68e6778a9f0df39109bb/docs/region_discovery.html" target="_blank">5b1d6da</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/e11b36d0fdef6278b31a1c6bae5287be77b06006/analysis/region_discovery.Rmd" target="_blank">e11b36d</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-09-16
</td>
<td>
workflowr::wflow_publish(“analysis/region_discovery.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/97ba74c28ccd521c9bb254b91a1fbea95abfb10a/docs/region_discovery.html" target="_blank">97ba74c</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-16
</td>
<td>
change math from 0.9 to 0.5 in html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/c85aa022256a7eff4f1d084b315cf9f6590423d5/docs/region_discovery.html" target="_blank">c85aa02</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/876c00781982b025b5a13944e06d2e7eae1225ec/analysis/region_discovery.Rmd" target="_blank">876c007</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-15
</td>
<td>
workflowr::wflow_publish(“analysis/region_discovery.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/adc95310b8280f0883dd6da680cb10b33ffa82aa/docs/region_discovery.html" target="_blank">adc9531</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/628c554e44d3a3049d0f7f827f944f3338fdbfc0/analysis/region_discovery.Rmd" target="_blank">628c554</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-15
</td>
<td>
workflowr::wflow_publish(“analysis/region_discovery.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/40fe3d314e901edeacabdb953349c58bcb5e6545/docs/region_discovery.html" target="_blank">40fe3d3</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/7137e73e88a2e2cc029e110addee22de065f5285/analysis/region_discovery.Rmd" target="_blank">7137e73</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-15
</td>
<td>
workflowr::wflow_publish(“analysis/region_discovery.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/d7ac25927d3d76dbd3381cc1faf8d1114c0de4a7/docs/region_discovery.html" target="_blank">d7ac259</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/1c105dac934dcd65bd9bf377e1bdccd4dd9eccc5/analysis/region_discovery.Rmd" target="_blank">1c105da</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-15
</td>
<td>
workflowr::wflow_publish(“analysis/region_discovery.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/a6edc2d304f7175669d209f776d3ebf46aad36ec/docs/region_discovery.html" target="_blank">a6edc2d</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/8089c88b02b3cec7d23e115b4a4924d802486234/docs/region_discovery.html" target="_blank">8089c88</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/1a427817be21230f81702a1b2ea6caaece02948f/analysis/region_discovery.Rmd" target="_blank">1a42781</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-14
</td>
<td>
workflowr::wflow_publish(“analysis/region_discovery.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/67ccd62e53ed2e69939ac53e315355c8643ef7c9/docs/region_discovery.html" target="_blank">67ccd62</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/1c6c90814c51d15c2f8b693c110380b0f58c56ea/analysis/region_discovery.Rmd" target="_blank">1c6c908</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-14
</td>
<td>
workflowr::wflow_publish(“analysis/region_discovery.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/a6ce03b5c924a9e06678a37cebd260c42944e2e0/docs/region_discovery.html" target="_blank">a6ce03b</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/d8b0308c11cb8e67f70a16e9b9b3c5f3857b6207/analysis/region_discovery.Rmd" target="_blank">d8b0308</a>
</td>
<td>
Nicholas Knoblauch
</td>
<td>
2019-09-14
</td>
<td>
workflowr::wflow_publish(“analysis/region_discovery.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/661bf21e91c9f54c6575f91cf36cd740bf17d633/docs/region_discovery.html" target="_blank">661bf21</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-09-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/ed68f10ad8c8b2e4adc0a4095c38b0e2cd717e73/analysis/region_discovery.Rmd" target="_blank">ed68f10</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-09-04
</td>
<td>
wflow_publish(“analysis/region_discovery.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CreRecombinase/ptb_workflowr/a6b09fbd05c0e3ceda8ac4cbfc1a7b0765fe6810/docs/region_discovery.html" target="_blank">a6b09fb</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-09-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CreRecombinase/ptb_workflowr/blob/30bd2f7212c14910169dcd08af4c65c63d985d55/analysis/region_discovery.Rmd" target="_blank">30bd2f7</a>
</td>
<td>
CreRecombinase
</td>
<td>
2019-09-04
</td>
<td>
wflow_publish(“analysis/region_discovery.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="locus-level-analysis" class="section level1">
<h1>Locus Level Analysis</h1>
<p>Below are the locus-level FDR for every region with an FDR of less than or equal to 0.1 in at least one of the models. There are 12 such regions in total</p>
<div id="htmlwidget-590d6b3f85a2bb8e6479" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-590d6b3f85a2bb8e6479">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12"],[614,356,15,373,353,973,1154,363,512,362,181,1592],[5,3,1,3,3,9,11,3,4,3,2,19],[156628700,126214943,21736588,154714218,121974097,79471208,101331121,139954597,174264132,137371083,73174848,16374416],[158825698,128194861,23086883,156008700,123517768,81079055,103959636,141339097,176570716,139954597,75630086,18409862],[2.39818e-19,3.67654e-07,3.30715e-06,0.000300154,0.00152026,0.00330151,0.0181833,0.0185725,0.0194758,0.023305,0.0363727,0.0644804],[2.39818e-19,4.87817e-07,3.60589e-06,0.000300154,0.00435247,0.00330151,0.0181833,0.0185725,0.0771672,0.023305,0.0657702,0.0644804],[3.67583e-19,3.82555e-07,4.49044e-06,0.00111824,0.00332695,0.00820667,0.0739324,0.0374278,0.0216398,0.0315783,0.0381889,0.104257],[3.68991e-19,3.67654e-07,4.5457e-06,0.00108618,0.0033962,0.00804852,0.0710849,0.0354023,0.0194758,0.0315941,0.0363727,0.104266],[6.52373e-19,7.12784e-07,3.30715e-06,0.00142276,0.00152026,0.0177934,0.265575,0.201817,0.18431,0.114235,0.126167,0.315027]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>region_id<\/th>\n      <th>chrom<\/th>\n      <th>start<\/th>\n      <th>stop<\/th>\n      <th>min_fdr<\/th>\n      <th>best<\/th>\n      <th>newfour<\/th>\n      <th>newnoeqtl<\/th>\n      <th>null<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="how-many-significant-regions-in-each-model" class="section level2">
<h2>How many significant regions in each model?</h2>
<p>For each model, the number of loci where FDR &lt; 0.05 (out of a total of 1703)</p>
<pre><code># A tibble: 4 x 2
  model     n_sig
  &lt;chr&gt;     &lt;int&gt;
1 newfour      10
2 newnoeqtl    10
3 best          9
4 null          6</code></pre>
<p>The <code>newfour</code> and <code>newnoeqtl</code> models both gave us 4 more loci than the <code>null</code> model. Neither of these models have negative multivariate effect estimates (besides the intercept). I’ll show results just for <code>newnoeqtl</code> from now on.</p>
</div>
<div id="what-are-the-new-regions" class="section level2">
<h2>What are the new regions?</h2>
<p>What are the four loci that have an FDR &lt; <code>0.05</code> under the <code>newnoeqtl</code>, but have an FDR above <code>0.05</code> under the null model? What is the gene</p>
</div>
<div id="regions-of-interest" class="section level2">
<h2>Regions of interest</h2>
<p>Xin was interested in the following SNPs because they had high p values but low pips with the null and high pips with the prior</p>
<div id="htmlwidget-c0b6d2dcd5110f82ecfd" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c0b6d2dcd5110f82ecfd">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7"],["18.0","356.0","373.0","512.0","614.0","614.0","973.0"],[1,3,3,4,5,5,9],[25796239,127878817,155868039,174729014,157884706,157884706,80455316],[7.385e-07,1.132e-12,5.041e-09,4.482e-07,3.036e-26,3.036e-26,1.449e-07],["rs138437971","rs2999049","rs4679761","rs7663453","rs2946164","rs2946164","rs1854325"],["chip-seq-pooled-DSC2-ctr-H3K4me1","chip-seq-pooled-DSC2-ctr-H3K4me1,hic_all_interacting_DT1_dTL4_D_48h","chip-seq-pooled-DSC2-ctr-H3K4me1,chip-seq-dec_up-H3K27ac","chip-seq-dec_up-H3K27ac,chip-seq-pooled-DSC2-ctr-H3K4me1,hic_all_interacting_DT1_dTL4_D_48h","","chip-seq-pooled-DSC2-ctr-H3K4me1","chip-seq-pooled-DSC2-ctr-H3K4me1,chip-seq-dec_up-H3K27ac"],["TMEM57","EEFSEC","KCNAB1","NBLA00301","EBF1","EBF1","GNAQ"],["Homo sapiens transmembrane protein 57 (TMEM57), mRNA.","Homo sapiens eukaryotic elongation factor, selenocysteine-tRNA-specific (EEFSEC), mRNA.","Homo sapiens potassium voltage-gated channel, shaker-related subfamily, beta member 1 (KCNAB1), transcript variant 1, mRNA.","Homo sapiens Nbla00301 (NBLA00301), non-coding RNA.","Homo sapiens early B-cell factor 1 (EBF1), mRNA.","Homo sapiens early B-cell factor 1 (EBF1), mRNA.","Homo sapiens guanine nucleotide binding protein (G protein), q polypeptide (GNAQ), mRNA."],[0.767422287973442,0.508729307959159,0.686159192990523,0.797488222426033,null,0.849287354512905,0.25968674972706],[0.776800791046683,0.515262858433021,0.684367675981605,0.800195048212099,null,0.830927229868519,0.30194576660095],[0.267181748909818,0.0666730825943379,0.0544273276315068,0.154399693606974,0.376192693636211,null,0.0353874946496127],[0.272109151779537,0.0681427815158344,0.05425943179788,0.163571190713554,0.320638877294739,null,0.0452571905677166]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>region_id<\/th>\n      <th>chrom<\/th>\n      <th>pos<\/th>\n      <th>p<\/th>\n      <th>rsid<\/th>\n      <th>features<\/th>\n      <th>geneSymbol<\/th>\n      <th>description<\/th>\n      <th>newnoeqtl_1<\/th>\n      <th>newnoeqtl_2<\/th>\n      <th>null_1<\/th>\n      <th>null_2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,9,10,11,12]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<!-- # Gene assignment of top `pip` SNPs -->
<!-- Below is a table with the nearest gene of SNPs with a `pip` $\gt 0.5$ for prior `newnoeqtl` and values of $L$: 1, 2 (max number of causal loci).  For each model I have taken the maximum `pip` over the values of $L$ (see the table below for each model-$L$ combination) -->
<!-- ```{r,echo=FALSE,message=FALSE,warning=FALSE} -->
<!-- all_pip_df <- dplyr::select(result_df,locusID=region_id,rsid=id,pos,chrom,p,pip,L,model) %>%  -->
<!--   filter(model %in% newmodel_df$name) %>%  -->
<!--   dplyr::group_by(model,rsid) %>%  -->
<!--   filter(pip == max(pip,na.rm = T)) %>%  -->
<!--   rename(max_pip = pip) %>%  -->
<!--   ungroup() %>%  -->
<!--   dplyr::select(-L) %>%  -->
<!--   spread(key=model,value=max_pip) %>% distinct() %>%  -->
<!--   nearest_gene_df(snpdf = .,dbdir=dbdir) %>% distinct() -->
<!-- ``` -->
<!-- ```{r,echo=FALSE,message=FALSE,warning=FALSE} -->
<!-- high_pip_df <- dplyr::select(result_df,locusID=region_id,rsid=id,pos,chrom,p,pip,L,model) %>%  -->
<!--   filter(model %in% newmodel_df$name) %>%  -->
<!--   dplyr::group_by(model,rsid) %>%  -->
<!--   filter(pip == max(pip,na.rm = T)) %>%  -->
<!--   rename(max_pip = pip) %>%  -->
<!--   ungroup() %>%  -->
<!--   dplyr::select(-L) %>%  -->
<!--   spread(key=model,value=max_pip) %>% distinct() %>%  -->
<!--   filter_at(vars(one_of(model_df$name)),any_vars(. > 0.5)) %>%  -->
<!--   nearest_gene_df(snpdf = .,dbdir=dbdir) -->
<!-- high_pip_df %>% mutate(prior_null_ratio=newnoeqtl/null) %>% left_join(anno_gw) %>%  -->
<!--   DT::datatable() -->
<!-- ``` -->
<!-- # Intersection with differential expression -->
<!-- I took the [ctr PL X dec TL](https://mnlab.uchicago.edu/mod/report/tcm/www-rna-seq-pooled/ctr.PL-ctr.TL.ruvseq/index.php) and [ctr PL X dec PL](https://mnlab.uchicago.edu/mod/report/tcm/www-rna-seq-pooled/ctr.TL-dec.TL.ruvseq/index.php) differential expression data and intersected it with the high-pip genes (`pip` $\gt 0.5$).  Below you can see the table of those results. (I filtered genes that did not have a $q$ value of $\let 0.05$ in at least one of the 2 experiments) -->
<!-- ```{r,echo = FALSE,message=FALSE,warning=FALSE} -->
<!-- library(httr) -->
<!-- library(rvest) -->
<!-- library(tidyverse) -->
<!-- user_pass <- yaml::read_yaml("data/ptb_cred.yml") -->
<!-- auth <- authenticate(user_pass$user, user_pass$pass,type = "digest") -->
<!-- diff_exp <- "https://mnlab.uchicago.edu/mod/rna-seq/E25-decidualization.14days/www-rna-seq-pooled-TL/" -->
<!-- exp_cmp <- c("ctr.PL-dec.PL", -->
<!--              "ctr.TL-dec.TL") -->
<!-- exp_df_f <- function(x){ -->
<!--   de_url <- glue("https://mnlab.uchicago.edu/mod/report/tcm/www-rna-seq-pooled/{x}.ruvseq/data/de.txt.zip") -->
<!--   de_df <- content(httr::GET(de_url,auth),"raw") -->
<!--   writeBin(de_df,"tmp.txt.gz") -->
<!--   read_tsv(unzip("tmp.txt.gz")) %>%  -->
<!--     rename(geneSymbol=gene) %>% dplyr::select(-ends_with("_TPM")) %>%  -->
<!--     mutate(treatment=x) -->
<!-- } -->
<!-- exp_df <- map_df(exp_cmp,exp_df_f) -->
<!-- # inp -->
<!-- # parse_expd <- function(x){ -->
<!-- #   pth <- url_parse(x)$path -->
<!-- #   over_under <- fs::path_file(fs::path_ext_remove(pth)) -->
<!-- #   trt <- stringr::str_replace(fs::path_split(pth)[[1]][6],"deseq2.(.+)X(.+).ruvseq","\\1_\\2") -->
<!-- #   read_tsv(content(httr::GET(x,auth),encoding = "UTF-8"),col_names="geneSymbol") %>% mutate(treatment=trt,over_under=over_under) -->
<!-- # } -->
<!-- #  -->
<!-- # diff_over <- content(httr::GET(diff_exp,auth),as = "parsed",encoding = "UTF-8") %>%  -->
<!-- #   html_node("table") %>%  -->
<!-- #   html_nodes(xpath = "//a[contains(@href,'genes.over.txt')]") %>%  -->
<!-- #   html_attr("href") %>%  -->
<!-- #   paste0(diff_exp,.) -->
<!-- #  -->
<!-- #  -->
<!-- # diff_under <- content(httr::GET(diff_exp,auth),as = "parsed",encoding = "UTF-8") %>%  -->
<!-- #   html_node("table") %>%  -->
<!-- #   html_nodes(xpath = "//a[contains(@href,'genes.under.txt')]") %>%  -->
<!-- #   html_attr("href") %>%  -->
<!-- #   paste0(diff_exp,.) -->
<!-- # alldiff_df <- map_df(c(diff_over, -->
<!-- #          diff_under),parse_expd) -->
<!-- exp_pip_df <- inner_join(exp_df,all_pip_df) -->
<!-- ``` -->
<!-- ```{r,echo=FALSE} -->
<!-- exp_pip_df %>%   -->
<!--   filter(best_pip>0.5,padj<0.05) %>%  -->
<!--   dplyr::select(-one_of(model_df$name)) %>%  -->
<!--   arrange(desc(best_pip)) %>%  -->
<!--   DT::datatable() -->
<!-- ``` -->
<!-- ## Assignment of top p-value SNPs -->
<!-- Below is a dataframe with the top SNP at every locus, and the gene that is nearest to the top SNP.  for each SNP I've included the `torus` locus discovery fdr from several models (best, newfour, newnoeqtl, null). -->
<!-- ```{r,echo=FALSE,message=FALSE,warning=FALSE} -->
<!--  fdr_table%>% DT::datatable() -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Manjaro Linux

Matrix products: default
BLAS/LAPACK: /usr/lib/libopenblas_haswellp-r0.3.6.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tidyselect_0.2.5   RSSp_0.9.0.9000    ldmap_0.0.0.9000  
 [4] daprcpp_1.0.0.9000 ldshrink_1.0-1     furrr_0.1.0.9002  
 [7] future_1.13.0      bigsnpr_0.11.5     bigstatsr_0.9.9   
[10] vroom_1.0.2.9000   RSQLite_2.1.1      drake_7.6.2.9000  
[13] fs_1.3.1           susieR_0.8.1.0525  here_0.1          
[16] dbplyr_1.4.2       MonetDBLite_0.6.1  glue_1.3.1        
[19] DT_0.7.2           forcats_0.4.0      stringr_1.4.0     
[22] dplyr_0.8.3        purrr_0.3.2.9000   readr_1.3.1       
[25] tidyr_0.8.99.9000  tibble_2.1.3       ggplot2_3.2.1.9000
[28] tidyverse_1.2.1   

loaded via a namespace (and not attached):
  [1] colorspace_1.4-1            RcppEigen_0.3.3.5.0        
  [3] ellipsis_0.2.0.9000         rprojroot_1.3-2            
  [5] XVector_0.24.0              GenomicRanges_1.36.0       
  [7] rstudioapi_0.10             listenv_0.7.0              
  [9] bit64_0.9-7                 fansi_0.4.0                
 [11] lubridate_1.7.4             xml2_1.2.2                 
 [13] codetools_0.2-16            knitr_1.23                 
 [15] zeallot_0.1.0               jsonlite_1.6               
 [17] workflowr_1.4.0             Rsamtools_2.0.0            
 [19] broom_0.5.2                 shiny_1.3.2                
 [21] compiler_3.6.1              httr_1.4.1                 
 [23] backports_1.1.4             assertthat_0.2.1           
 [25] Matrix_1.2-17               cli_1.1.0                  
 [27] later_0.8.0                 htmltools_0.3.6            
 [29] tools_3.6.1                 igraph_1.2.4.1             
 [31] gtable_0.3.0                GenomeInfoDbData_1.2.1     
 [33] Rcpp_1.0.2                  Biobase_2.44.0             
 [35] cellranger_1.1.0            Biostrings_2.52.0          
 [37] vctrs_0.2.0.9002            nlme_3.1-140               
 [39] rtracklayer_1.44.0          crosstalk_1.0.0            
 [41] iterators_1.0.10            wavethresh_4.6.8           
 [43] xfun_0.8                    globals_0.12.4             
 [45] plyranges_1.4.3             rvest_0.3.4                
 [47] mime_0.7                    lifecycle_0.1.0            
 [49] XML_3.98-1.20               zlibbioc_1.30.0            
 [51] MASS_7.3-51.4               scales_1.0.0               
 [53] promises_1.0.1              hms_0.5.1                  
 [55] parallel_3.6.1              SummarizedExperiment_1.14.0
 [57] yaml_2.2.0                  memoise_1.1.0              
 [59] stringi_1.4.3               S4Vectors_0.22.0           
 [61] foreach_1.4.4               BiocGenerics_0.30.0        
 [63] filelock_1.0.2              BiocParallel_1.18.0        
 [65] storr_1.2.2                 GenomeInfoDb_1.20.0        
 [67] matrixStats_0.54.0          rlang_0.4.0.9002           
 [69] pkgconfig_2.0.2             bitops_1.0-6               
 [71] evaluate_0.14               lattice_0.20-38            
 [73] GenomicAlignments_1.20.0    htmlwidgets_1.3            
 [75] cowplot_1.0.0               bit_1.1-14                 
 [77] magrittr_1.5                R6_2.4.0                   
 [79] IRanges_2.18.1              generics_0.0.2             
 [81] base64url_1.4               DelayedArray_0.10.0        
 [83] txtq_0.1.6                  DBI_1.0.0                  
 [85] pillar_1.4.2                haven_2.1.0                
 [87] whisker_0.4                 withr_2.1.2                
 [89] RCurl_1.95-4.12             modelr_0.1.4               
 [91] crayon_1.3.4                utf8_1.1.4                 
 [93] rmarkdown_1.13              grid_3.6.1                 
 [95] readxl_1.3.1                data.table_1.12.2          
 [97] blob_1.1.1                  git2r_0.26.1               
 [99] digest_0.6.20               xtable_1.8-4               
[101] httpuv_1.5.1                RcppParallel_4.4.3         
[103] stats4_3.6.1                munsell_0.5.0              </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
