---
title: "Preliminary Enrichment Fine-mapping Results"
author: "Nicholas Knoblauch"
date: "2019-07-02"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---
  
  
```{r setup, message=FALSE, warning=FALSE, include=FALSE}
knitr::opts_chunk$set(
    dev = c("svg"),
)

```  




```{r}


```



```{r,echo=F,message=FALSE,warning=FALSE}
library(DT)
source("R/config.R")
source("R/plan.R")   
cc <- drake_cache("~/Dropbox/Repos/cache/desktop")
loadd(ngwas_i,cache=cc)
snp_range <-  group_by(ngwas_i,region_id) %>% 
  summarise(snpct=n())
```


```{r}
exp_reg <- function(df){
  glue::glue_data(df,"{chrom}:{start}-{stop}")
}
```


```{r}
fn <- "/home/nwknoblauch/tmp/bcf/ALL.chr19.phase3_shapeit2_mvncall_integrated_v5.20130502.genotypes.bcf"
bfn <- "/home/nwknoblauch/tmp/ALL.chr19.phase3_shapeit2_mvncall_integrated_v5.20130502.genotypes.gds"
library(SeqArray)

trn <- dplyr::filter(ld_df,chrom==19) %>% slice(2) %>% exp_reg
tret <- readVCFToListByRange(fn,range = trn)
fn
```


## The loci

The genome was broken in to `r nrow(snp_range)` regions (using Pickrell's `ldetect`).  Regions were broken into subregions if they contained greater than `r max(snp_range$snpct)-min(snp_range$snpct)` SNPs, unless this would result in a subregion with fewer than `50` SNPs.  This means that  region size varied from `r min(snp_range$snpct)` SNPs to `max(snp_range$snpct)` SNPs. The functional enrichment program `torus` was run individually on each of the following features:

```{r,echo=FALSE}
all_feat_df <- map_df(str_replace_all(all_feat,"\\-","\\."),~readd(paste0("s_feat_anno_r_ra_",.x),character_only = T,cache = cc)$est)

filter(all_feat_df,term!="Intercept") %>% DT::datatable()
```

## Forward Selection

Four steps of forward selection were used to obtain a final multivariate model. In each step, the term that added the most to the joint likelihood was added to the model.


```{r,echo=FALSE,message=FALSE}
loadd(feat4,cache=cc)
loadd(prior_r,cache=cc)
loadd(top_gwas_reg,cache=cc)

feature_df <- tibble(feature=feat4)
DT::datatable(feature_df)
```


Fine mapping with `susie` was then run on the top 5 loci, after generating priors using `torus`.  `susie` was run assuming 1 causal variant per locus.

## Locus 1

The top hit is on chromosome 1.  

```{r,echo=FALSE,message=FALSE}

all_susie_r <- map2_dfr(1:15,prior_r,function(x,y){
  mutate(y,pip=readd(glue::glue("susie_res_{x}L"),character_only = T,cache=cc)$pip)
})

susie_res_df <- dplyr::select(all_susie_r,chrom,pos,z=`z-stat`,region_id,prior,pip) %>% tidyr::gather(key="feature",value="value",z,prior,pip)

dplyr::filter(susie_res_df,region_id==region_id[1]) %>% ggplot(aes(x=pos,y=abs(value),col=feature))+geom_point()+facet_wrap(~feature,scales = "free_y")+ggtitle("Chromosome 1,76481857-76728135")
```

