---
title: "region_discovery"
author: "CreRecombinase"
date: "2019-09-04"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


```{r,echo=FALSE,warning=FALSE,message=FALSE}
library(tidyverse)
library(glue)
source("R/config.R")
L <- c(1L,2L,3L)
names_d <- model_df$name
model_terms <- model_df$features
names(model_terms) <- model_df$name
all_feat <- unique(c(unlist(model_terms)))
cc <- drake::drake_cache("cache/rcc4/")
dbdir <-"/home/nwknoblauch/Dropbox/scratch/ptb_scratch/gene_anno/"
drake::loadd(top_gwas_loc,cache=cc)
drake::loadd(top_gwas_reg,cache=cc)
top_regions <- top_gwas_reg$region_id
top_snp <- dplyr::select(top_gwas_loc,rsid=id,chrom,pos,min_p=p,region_id=region_id) %>% dplyr::mutate(region_id=as.integer(as.numeric(region_id))) %>%
  group_by(region_id) %>% filter(min_p==min(min_p)) %>% 
  ungroup() %>% 
  nearest_gene_df(snpdf = .,dbdir=dbdir,needs_mRNA = TRUE)
fdr_results <- map_df(names_d,function(x){
    mutate(drake::readd(glue("fdr_results_.scratch.midway2.nwknoblauch.ptb_scratch.{x}.txt.gz"),character_only = T,cache=cc),model=x)
}) %>% dplyr::select(region_id,fdr,model) %>% inner_join(data_config$data$ld_df) %>% inner_join(top_snp)

fdr_table <- spread(fdr_results,key=model,value=fdr) %>% arrange(min_p)

result_df <- cross_df(list(L=L,
                           locus=seq_len(nrow(top_gwas_reg)),
                           names=names_d)) %>% 
  pmap_df(function(L,locus,names){
    dplyr::mutate(drake::readd(glue("susie_res_{locus}L_{L}L_prior_r_mix_results_.scratch.midway2.nwknoblauch.ptb_scratch.{names}.txt.gz"),character_only = T,cache=cc)$df,L=L,locus=locus,model=names)
})

```


# Gene assignment of top `pip` SNPs


Below is a table with the nearest gene of SNPs with a `pip` $\gt 0.5$ for _any_ combinations of priors and values of $L$ (max number of causal loci).  For each model I have taken the maximum `pip` over the three values of $L$ (see the table below for each model-$L$ combination)


```{r,echo=FALSE,message=FALSE,warning=FALSE}
all_pip_df <- dplyr::select(result_df,locusID=region_id,rsid=id,pos,chrom,p,pip,L,model) %>% 
  dplyr::group_by(model,rsid) %>% 
  filter(pip==max(pip,na.rm = T)) %>% 
  rename(max_pip=pip) %>% 
  ungroup() %>% group_by(rsid) %>% mutate(best_pip=max(max_pip)) %>% ungroup() %>% 
  dplyr::select(-L) %>% 
  spread(key=model,value=max_pip) %>% 
  nearest_gene_df(snpdf = .,dbdir=dbdir)
```



```{r,echo=FALSE,message=FALSE,warning=FALSE}
high_pip_df <- dplyr::select(result_df,locusID=region_id,rsid=id,pos,chrom,p,pip,L,model) %>% 
  dplyr::group_by(model,rsid) %>% 
  filter(pip==max(pip,na.rm = T)) %>% 
  rename(max_pip=pip) %>% 
  ungroup() %>% 
  dplyr::select(-L) %>% 
  spread(key=model,value=max_pip) %>% 
  filter_at(vars(one_of(model_df$name)),any_vars(. > 0.5)) %>% 
  nearest_gene_df(snpdf = .,dbdir=dbdir)

high_pip_df %>% 
  DT::datatable()
```




# Intersection with differential expression


I took the [ctr PL X dec TL](https://mnlab.uchicago.edu/mod/report/tcm/www-rna-seq-pooled/ctr.PL-ctr.TL.ruvseq/index.php) and [ctr PL X dec PL](https://mnlab.uchicago.edu/mod/report/tcm/www-rna-seq-pooled/ctr.TL-dec.TL.ruvseq/index.php) differential expression data and intersected it with the high-pip genes (`pip` $\gt 0.5$).  Below you can see the table of those results. (I filtered genes that did not have a $q$ value of $\let 0.05$ in at least one of the 2 experiments)


```{r,echo = FALSE,message=FALSE,warning=FALSE}
library(httr)
library(rvest)
library(tidyverse)
user_pass <- yaml::read_yaml("data/ptb_cred.yml")
auth <- authenticate(user_pass$user, user_pass$pass,type = "digest")
diff_exp <- "https://mnlab.uchicago.edu/mod/rna-seq/E25-decidualization.14days/www-rna-seq-pooled-TL/"
exp_cmp <- c("ctr.PL-dec.PL",
             "ctr.TL-dec.TL")

exp_df_f <- function(x){
  de_url <- glue("https://mnlab.uchicago.edu/mod/report/tcm/www-rna-seq-pooled/{x}.ruvseq/data/de.txt.zip")
  de_df <- content(httr::GET(de_url,auth),"raw")
  writeBin(de_df,"tmp.txt.gz")
  read_tsv(unzip("tmp.txt.gz")) %>% 
    rename(geneSymbol=gene) %>% dplyr::select(-ends_with("_TPM")) %>% 
    mutate(treatment=x)
}
exp_df <- map_df(exp_cmp,exp_df_f)
# inp
# parse_expd <- function(x){
#   pth <- url_parse(x)$path
#   over_under <- fs::path_file(fs::path_ext_remove(pth))
#   trt <- stringr::str_replace(fs::path_split(pth)[[1]][6],"deseq2.(.+)X(.+).ruvseq","\\1_\\2")
#   read_tsv(content(httr::GET(x,auth),encoding = "UTF-8"),col_names="geneSymbol") %>% mutate(treatment=trt,over_under=over_under)
# }
# 
# diff_over <- content(httr::GET(diff_exp,auth),as = "parsed",encoding = "UTF-8") %>% 
#   html_node("table") %>% 
#   html_nodes(xpath = "//a[contains(@href,'genes.over.txt')]") %>% 
#   html_attr("href") %>% 
#   paste0(diff_exp,.)
# 
# 
# diff_under <- content(httr::GET(diff_exp,auth),as = "parsed",encoding = "UTF-8") %>% 
#   html_node("table") %>% 
#   html_nodes(xpath = "//a[contains(@href,'genes.under.txt')]") %>% 
#   html_attr("href") %>% 
#   paste0(diff_exp,.)


# alldiff_df <- map_df(c(diff_over,
#          diff_under),parse_expd)

exp_pip_df <- inner_join(exp_df,all_pip_df)

```


```{r,echo=FALSE}
exp_pip_df %>%  
  filter(best_pip>0.5,padj<0.05) %>% 
  dplyr::select(-one_of(model_df$name)) %>% 
  arrange(desc(best_pip)) %>% 
  DT::datatable()
```




## Assignment of top p-value SNPs

Below is a dataframe with the top SNP at every locus, and the gene that is nearest to the top SNP.  for each SNP I've included the `torus` locus discovery fdr from several models (`r model_df$name`).
```{r,echo=FALSE,message=FALSE,warning=FALSE}
 fdr_table%>% DT::datatable()
```




